From 7de648b59a8c179bfa6f65b6a1b76eba25a811a8 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Thu, 10 May 2018 09:22:20 -0500
Subject: [PATCH 3801/5242] MLK-18085 camera: Change the OV10635 to use 96 MHz
 timing.

commit  902f497c5ced806438d35fd114069a63ea183e33 from
https://source.codeaurora.org/external/imx/linux-imx.git

This issue changes the OV10635 to use the 96 MHz timing. In the one camera the configuration, the 72MHz timing was causing vsync and hsync errors. The 96 MHz timing will work for any number of cameras.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/max9286.c | 2015 ++++++++++++++++-----------------
 1 file changed, 976 insertions(+), 1039 deletions(-)

diff --git a/drivers/media/platform/imx8/max9286.c b/drivers/media/platform/imx8/max9286.c
index 3f9baa4..64d79a0 100644
--- a/drivers/media/platform/imx8/max9286.c
+++ b/drivers/media/platform/imx8/max9286.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 NXP
+ * Copyright 2017-2018 NXP
  */
 /*
  * The code contained herein is licensed under the GNU General Public
@@ -162,7 +162,7 @@ enum ov10635_frame_rate {
 	{ 0x3503, 0x10, 0 },
 	{ 0x3025, 0x03, 0 },
 	{ 0x3003, 0x20, 0 },
-	{ 0x3004, 0x23, 0 },
+	{ 0x3004, 0x21, 0 },
 	{ 0x3005, 0x20, 0 },
 	{ 0x3006, 0x91, 0 },
 	{ 0x3600, 0x74, 0 },
@@ -176,13 +176,13 @@ enum ov10635_frame_rate {
 	{ 0x3631, 0x16, 0 },
 	{ 0x3714, 0x10, 0 },
 	{ 0x371d, 0x01, 0 },
-	{ 0x4300, 0x38, 0},
+	{ 0x4300, 0x38, 0 },
 	{ 0x3007, 0x01, 0 },
-	{ 0x3024, 0x01, 0 },  /* YUV */
+	{ 0x3024, 0x01, 0 },
 	{ 0x3020, 0x0b, 0 },
-	{ 0x3702, 0x10, 0 },  /* 0x20 for 96MHz */
-	{ 0x3703, 0x24, 0 },  /* 0x48 for 96MHz */
-	{ 0x3704, 0x19, 0 },
+	{ 0x3702, 0x20, 0 },
+	{ 0x3703, 0x48, 0 },
+	{ 0x3704, 0x32, 0 },
 	{ 0x3709, 0xa8, 0 },
 	{ 0x3709, 0xa8, 0 },
 	{ 0x370c, 0xc7, 0 },
@@ -420,8 +420,8 @@ enum ov10635_frame_rate {
 	{ 0x405a, 0x00, 0 },
 	{ 0x4202, 0x02, 0 },
 	{ 0x3023, 0x10, 0 },
-	{ 0x0100, 0x01, 0 },
-	{ 0x0100, 0x01, 0 },
+	{ 0x100 , 0x0f, 0 },
+	{ 0x100 , 0x0f, 0 },
 	{ 0x6f10, 0x07, 0 },
 	{ 0x6f11, 0x82, 0 },
 	{ 0x6f12, 0x04, 0 },
@@ -665,135 +665,135 @@ enum ov10635_frame_rate {
 	{ 0xd0e0, 0xa8, 0 },
 	{ 0xd0e1, 0xe3, 0 },
 	{ 0xd0e2, 0x40, 0 },
-	{ 0xd0e3, 0x08, 0},
+	{ 0xd0e3, 0x08, 0 },
 	{ 0xd0e4, 0x8c, 0 },
-	{ 0xd0e5, 0x84, 0},
+	{ 0xd0e5, 0x84, 0 },
 	{ 0xd0e6, 0x00, 0 },
 	{ 0xd0e7, 0x00, 0 },
-	{ 0xd0e8, 0xa8, 0},
-	{ 0xd0e9, 0xa3, 0},
-	{ 0xd0ea, 0x40, 0},
-	{ 0xd0eb, 0x09, 0},
-	{ 0xd0ec, 0xa8, 0},
-	{ 0xd0ed, 0xc3, 0},
-	{ 0xd0ee, 0x38, 0},
-	{ 0xd0ef, 0x2a, 0},
-	{ 0xd0f0, 0xd8, 0},
-	{ 0xd0f1, 0x07, 0},
-	{ 0xd0f2, 0x20, 0},
+	{ 0xd0e8, 0xa8, 0 },
+	{ 0xd0e9, 0xa3, 0 },
+	{ 0xd0ea, 0x40, 0 },
+	{ 0xd0eb, 0x09, 0 },
+	{ 0xd0ec, 0xa8, 0 },
+	{ 0xd0ed, 0xc3, 0 },
+	{ 0xd0ee, 0x38, 0 },
+	{ 0xd0ef, 0x2a, 0 },
+	{ 0xd0f0, 0xd8, 0 },
+	{ 0xd0f1, 0x07, 0 },
+	{ 0xd0f2, 0x20, 0 },
 	{ 0xd0f3, 0x00, 0 },
-	{ 0xd0f4, 0x8c, 0},
-	{ 0xd0f5, 0x66, 0},
+	{ 0xd0f4, 0x8c, 0 },
+	{ 0xd0f5, 0x66, 0 },
 	{ 0xd0f6, 0x00, 0 },
-	{ 0xd0f7, 0x00, 0},
-	{ 0xd0f8, 0xd8, 0},
-	{ 0xd0f9, 0x05, 0},
-	{ 0xd0fa, 0x18, 0},
-	{ 0xd0fb, 0x00, 0},
-	{ 0xd0fc, 0x18, 0},
-	{ 0xd0fd, 0x60, 0},
+	{ 0xd0f7, 0x00, 0 },
+	{ 0xd0f8, 0xd8, 0 },
+	{ 0xd0f9, 0x05, 0 },
+	{ 0xd0fa, 0x18, 0 },
+	{ 0xd0fb, 0x00, 0 },
+	{ 0xd0fc, 0x18, 0 },
+	{ 0xd0fd, 0x60, 0 },
 	{ 0xd0fe, 0x00, 0 },
-	{ 0xd0ff, 0x01, 0},
-	{ 0xd100, 0x98, 0},
-	{ 0xd101, 0x90, 0},
+	{ 0xd0ff, 0x01, 0 },
+	{ 0xd100, 0x98, 0 },
+	{ 0xd101, 0x90, 0 },
 	{ 0xd102, 0x00, 0 },
 	{ 0xd103, 0x00, 0 },
-	{ 0xd104, 0x84, 0},
-	{ 0xd105, 0xae, 0},
-	{ 0xd106, 0x00, 0},
-	{ 0xd107, 0x00, 0},
-	{ 0xd108, 0xa8, 0},
-	{ 0xd109, 0x63, 0},
-	{ 0xd10a, 0x06, 0},
-	{ 0xd10b, 0x4c, 0},
-	{ 0xd10c, 0x9c, 0},
-	{ 0xd10d, 0xc0, 0},
+	{ 0xd104, 0x84, 0 },
+	{ 0xd105, 0xae, 0 },
+	{ 0xd106, 0x00, 0 },
+	{ 0xd107, 0x00, 0 },
+	{ 0xd108, 0xa8, 0 },
+	{ 0xd109, 0x63, 0 },
+	{ 0xd10a, 0x06, 0 },
+	{ 0xd10b, 0x4c, 0 },
+	{ 0xd10c, 0x9c, 0 },
+	{ 0xd10d, 0xc0, 0 },
 	{ 0xd10e, 0x00, 0 },
 	{ 0xd10f, 0x00, 0 },
-	{ 0xd110, 0xd8, 0},
-	{ 0xd111, 0x03, 0},
-	{ 0xd112, 0x30, 0},
-	{ 0xd113, 0x00, 0},
-	{ 0xd114, 0x8c, 0},
-	{ 0xd115, 0x65, 0},
-	{ 0xd116, 0x00, 0},
-	{ 0xd117, 0x6e, 0},
-	{ 0xd118, 0xe5, 0},
-	{ 0xd119, 0x84, 0},
+	{ 0xd110, 0xd8, 0 },
+	{ 0xd111, 0x03, 0 },
+	{ 0xd112, 0x30, 0 },
+	{ 0xd113, 0x00, 0 },
+	{ 0xd114, 0x8c, 0 },
+	{ 0xd115, 0x65, 0 },
+	{ 0xd116, 0x00, 0 },
+	{ 0xd117, 0x6e, 0 },
+	{ 0xd118, 0xe5, 0 },
+	{ 0xd119, 0x84, 0 },
 	{ 0xd11a, 0x18, 0 },
 	{ 0xd11b, 0x00, 0 },
-	{ 0xd11c, 0x10, 0},
-	{ 0xd11d, 0x00, 0},
-	{ 0xd11e, 0x00, 0},
-	{ 0xd11f, 0x07, 0},
-	{ 0xd120, 0x18, 0},
-	{ 0xd121, 0x80, 0},
-	{ 0xd122, 0x80, 0},
-	{ 0xd123, 0x06, 0},
-	{ 0xd124, 0x94, 0},
-	{ 0xd125, 0x65, 0},
-	{ 0xd126, 0x00, 0},
-	{ 0xd127, 0x70, 0},
-	{ 0xd128, 0xe5, 0},
-	{ 0xd129, 0x43, 0},
-	{ 0xd12a, 0x60, 0},
+	{ 0xd11c, 0x10, 0 },
+	{ 0xd11d, 0x00, 0 },
+	{ 0xd11e, 0x00, 0 },
+	{ 0xd11f, 0x07, 0 },
+	{ 0xd120, 0x18, 0 },
+	{ 0xd121, 0x80, 0 },
+	{ 0xd122, 0x80, 0 },
+	{ 0xd123, 0x06, 0 },
+	{ 0xd124, 0x94, 0 },
+	{ 0xd125, 0x65, 0 },
+	{ 0xd126, 0x00, 0 },
+	{ 0xd127, 0x70, 0 },
+	{ 0xd128, 0xe5, 0 },
+	{ 0xd129, 0x43, 0 },
+	{ 0xd12a, 0x60, 0 },
 	{ 0xd12b, 0x00, 0 },
-	{ 0xd12c, 0x0c, 0},
-	{ 0xd12d, 0x00, 0},
+	{ 0xd12c, 0x0c, 0 },
+	{ 0xd12d, 0x00, 0 },
 	{ 0xd12e, 0x00, 0 },
-	{ 0xd12f, 0x3e, 0},
-	{ 0xd130, 0xa8, 0},
-	{ 0xd131, 0x64, 0},
-	{ 0xd132, 0x38, 0},
-	{ 0xd133, 0x24, 0},
-	{ 0xd134, 0x18, 0},
+	{ 0xd12f, 0x3e, 0 },
+	{ 0xd130, 0xa8, 0 },
+	{ 0xd131, 0x64, 0 },
+	{ 0xd132, 0x38, 0 },
+	{ 0xd133, 0x24, 0 },
+	{ 0xd134, 0x18, 0 },
 	{ 0xd135, 0x80, 0 },
-	{ 0xd136, 0x80, 0},
-	{ 0xd137, 0x06, 0},
+	{ 0xd136, 0x80, 0 },
+	{ 0xd137, 0x06, 0 },
 	{ 0xd138, 0xa8, 0 },
-	{ 0xd139, 0x64, 0},
+	{ 0xd139, 0x64, 0 },
 	{ 0xd13a, 0x38, 0 },
-	{ 0xd13b, 0x24, 0},
-	{ 0xd13c, 0x8c, 0},
-	{ 0xd13d, 0x63, 0},
-	{ 0xd13e, 0x00, 0},
-	{ 0xd13f, 0x00, 0},
-	{ 0xd140, 0xa4, 0},
-	{ 0xd141, 0x63, 0},
-	{ 0xd142, 0x00, 0},
-	{ 0xd143, 0x40, 0},
-	{ 0xd144, 0xbc, 0},
-	{ 0xd145, 0x23, 0},
-	{ 0xd146, 0x00, 0},
-	{ 0xd147, 0x00, 0},
-	{ 0xd148, 0x0c, 0},
-	{ 0xd149, 0x00, 0},
+	{ 0xd13b, 0x24, 0 },
+	{ 0xd13c, 0x8c, 0 },
+	{ 0xd13d, 0x63, 0 },
+	{ 0xd13e, 0x00, 0 },
+	{ 0xd13f, 0x00, 0 },
+	{ 0xd140, 0xa4, 0 },
+	{ 0xd141, 0x63, 0 },
+	{ 0xd142, 0x00, 0 },
+	{ 0xd143, 0x40, 0 },
+	{ 0xd144, 0xbc, 0 },
+	{ 0xd145, 0x23, 0 },
+	{ 0xd146, 0x00, 0 },
+	{ 0xd147, 0x00, 0 },
+	{ 0xd148, 0x0c, 0 },
+	{ 0xd149, 0x00, 0 },
 	{ 0xd14a, 0x00, 0 },
-	{ 0xd14b, 0x2a, 0},
-	{ 0xd14c, 0xa8, 0},
-	{ 0xd14d, 0x64, 0},
-	{ 0xd14e, 0x6e, 0},
-	{ 0xd14f, 0x44, 0},
-	{ 0xd150, 0x19, 0},
-	{ 0xd151, 0x00, 0},
-	{ 0xd152, 0x80, 0},
-	{ 0xd153, 0x06, 0},
-	{ 0xd154, 0xa8, 0},
-	{ 0xd155, 0xe8, 0},
-	{ 0xd156, 0x3d, 0},
-	{ 0xd157, 0x05, 0},
-	{ 0xd158, 0x8c, 0},
-	{ 0xd159, 0x67, 0},
+	{ 0xd14b, 0x2a, 0 },
+	{ 0xd14c, 0xa8, 0 },
+	{ 0xd14d, 0x64, 0 },
+	{ 0xd14e, 0x6e, 0 },
+	{ 0xd14f, 0x44, 0 },
+	{ 0xd150, 0x19, 0 },
+	{ 0xd151, 0x00, 0 },
+	{ 0xd152, 0x80, 0 },
+	{ 0xd153, 0x06, 0 },
+	{ 0xd154, 0xa8, 0 },
+	{ 0xd155, 0xe8, 0 },
+	{ 0xd156, 0x3d, 0 },
+	{ 0xd157, 0x05, 0 },
+	{ 0xd158, 0x8c, 0 },
+	{ 0xd159, 0x67, 0 },
 	{ 0xd15a, 0x00, 0 },
-	{ 0xd15b, 0x00, 0},
-	{ 0xd15c, 0xb8, 0},
-	{ 0xd15d, 0x63, 0},
+	{ 0xd15b, 0x00, 0 },
+	{ 0xd15c, 0xb8, 0 },
+	{ 0xd15d, 0x63, 0 },
 	{ 0xd15e, 0x00, 0 },
-	{ 0xd15f, 0x18, 0},
-	{ 0xd160, 0xb8, 0},
-	{ 0xd161, 0x63, 0},
+	{ 0xd15f, 0x18, 0 },
+	{ 0xd160, 0xb8, 0 },
+	{ 0xd161, 0x63, 0 },
 	{ 0xd162, 0x00, 0 },
-	{ 0xd163, 0x98, 0},
+	{ 0xd163, 0x98, 0 },
 	{ 0xd164, 0xbc, 0 },
 	{ 0xd165, 0x03, 0 },
 	{ 0xd166, 0x00, 0 },
@@ -801,927 +801,927 @@ enum ov10635_frame_rate {
 	{ 0xd168, 0x10, 0 },
 	{ 0xd169, 0x00, 0 },
 	{ 0xd16a, 0x00, 0 },
-	{ 0xd16b, 0x10, 0},
-	{ 0xd16c, 0xa9, 0},
-	{ 0xd16d, 0x48, 0},
-	{ 0xd16e, 0x67, 0},
+	{ 0xd16b, 0x10, 0 },
+	{ 0xd16c, 0xa9, 0 },
+	{ 0xd16d, 0x48, 0 },
+	{ 0xd16e, 0x67, 0 },
 	{ 0xd16f, 0x02, 0 },
-	{ 0xd170, 0xb8, 0},
-	{ 0xd171, 0xa3, 0},
+	{ 0xd170, 0xb8, 0 },
+	{ 0xd171, 0xa3, 0 },
 	{ 0xd172, 0x00, 0 },
-	{ 0xd173, 0x19, 0},
+	{ 0xd173, 0x19, 0 },
 	{ 0xd174, 0x8c, 0 },
-	{ 0xd175, 0x8a, 0},
+	{ 0xd175, 0x8a, 0 },
 	{ 0xd176, 0x00, 0 },
 	{ 0xd177, 0x00, 0 },
-	{ 0xd178, 0xa9, 0},
-	{ 0xd179, 0x68, 0},
-	{ 0xd17a, 0x67, 0},
-	{ 0xd17b, 0x03, 0},
-	{ 0xd17c, 0xb8, 0},
-	{ 0xd17d, 0xc4, 0},
+	{ 0xd178, 0xa9, 0 },
+	{ 0xd179, 0x68, 0 },
+	{ 0xd17a, 0x67, 0 },
+	{ 0xd17b, 0x03, 0 },
+	{ 0xd17c, 0xb8, 0 },
+	{ 0xd17d, 0xc4, 0 },
 	{ 0xd17e, 0x00, 0 },
-	{ 0xd17f, 0x08, 0},
-	{ 0xd180, 0x8c, 0},
-	{ 0xd181, 0x6b, 0},
-	{ 0xd182, 0x00, 0},
-	{ 0xd183, 0x00, 0},
-	{ 0xd184, 0xb8, 0},
-	{ 0xd185, 0x85, 0},
+	{ 0xd17f, 0x08, 0 },
+	{ 0xd180, 0x8c, 0 },
+	{ 0xd181, 0x6b, 0 },
+	{ 0xd182, 0x00, 0 },
+	{ 0xd183, 0x00, 0 },
+	{ 0xd184, 0xb8, 0 },
+	{ 0xd185, 0x85, 0 },
 	{ 0xd186, 0x00, 0 },
-	{ 0xd187, 0x98, 0},
-	{ 0xd188, 0xe0, 0},
-	{ 0xd189, 0x63, 0},
-	{ 0xd18a, 0x30, 0},
-	{ 0xd18b, 0x04, 0},
-	{ 0xd18c, 0xe0, 0},
-	{ 0xd18d, 0x64, 0},
-	{ 0xd18e, 0x18, 0},
-	{ 0xd18f, 0x00, 0},
-	{ 0xd190, 0xa4, 0},
-	{ 0xd191, 0x83, 0},
-	{ 0xd192, 0xff, 0},
-	{ 0xd193, 0xff, 0},
-	{ 0xd194, 0xb8, 0},
-	{ 0xd195, 0x64, 0},
+	{ 0xd187, 0x98, 0 },
+	{ 0xd188, 0xe0, 0 },
+	{ 0xd189, 0x63, 0 },
+	{ 0xd18a, 0x30, 0 },
+	{ 0xd18b, 0x04, 0 },
+	{ 0xd18c, 0xe0, 0 },
+	{ 0xd18d, 0x64, 0 },
+	{ 0xd18e, 0x18, 0 },
+	{ 0xd18f, 0x00, 0 },
+	{ 0xd190, 0xa4, 0 },
+	{ 0xd191, 0x83, 0 },
+	{ 0xd192, 0xff, 0 },
+	{ 0xd193, 0xff, 0 },
+	{ 0xd194, 0xb8, 0 },
+	{ 0xd195, 0x64, 0 },
 	{ 0xd196, 0x00, 0 },
-	{ 0xd197, 0x48, 0},
-	{ 0xd198, 0xd8, 0},
-	{ 0xd199, 0x0a, 0},
-	{ 0xd19a, 0x18, 0},
-	{ 0xd19b, 0x00, 0},
-	{ 0xd19c, 0xd8, 0},
-	{ 0xd19d, 0x0b, 0},
-	{ 0xd19e, 0x20, 0},
-	{ 0xd19f, 0x00, 0},
-	{ 0xd1a0, 0x9c, 0},
-	{ 0xd1a1, 0x60, 0},
-	{ 0xd1a2, 0x00, 0},
-	{ 0xd1a3, 0x00, 0},
-	{ 0xd1a4, 0xd8, 0},
-	{ 0xd1a5, 0x07, 0},
-	{ 0xd1a6, 0x18, 0},
+	{ 0xd197, 0x48, 0 },
+	{ 0xd198, 0xd8, 0 },
+	{ 0xd199, 0x0a, 0 },
+	{ 0xd19a, 0x18, 0 },
+	{ 0xd19b, 0x00, 0 },
+	{ 0xd19c, 0xd8, 0 },
+	{ 0xd19d, 0x0b, 0 },
+	{ 0xd19e, 0x20, 0 },
+	{ 0xd19f, 0x00, 0 },
+	{ 0xd1a0, 0x9c, 0 },
+	{ 0xd1a1, 0x60, 0 },
+	{ 0xd1a2, 0x00, 0 },
+	{ 0xd1a3, 0x00, 0 },
+	{ 0xd1a4, 0xd8, 0 },
+	{ 0xd1a5, 0x07, 0 },
+	{ 0xd1a6, 0x18, 0 },
 	{ 0xd1a7, 0x00, 0 },
 	{ 0xd1a8, 0xa8, 0 },
-	{ 0xd1a9, 0x68, 0},
+	{ 0xd1a9, 0x68, 0 },
 	{ 0xd1aa, 0x38, 0 },
-	{ 0xd1ab, 0x22, 0},
-	{ 0xd1ac, 0x9c, 0},
-	{ 0xd1ad, 0x80, 0},
-	{ 0xd1ae, 0x00, 0},
-	{ 0xd1af, 0x70, 0},
-	{ 0xd1b0, 0xa8, 0},
-	{ 0xd1b1, 0xe8, 0},
-	{ 0xd1b2, 0x38, 0},
-	{ 0xd1b3, 0x43, 0},
-	{ 0xd1b4, 0xd8, 0},
-	{ 0xd1b5, 0x03, 0},
-	{ 0xd1b6, 0x20, 0},
+	{ 0xd1ab, 0x22, 0 },
+	{ 0xd1ac, 0x9c, 0 },
+	{ 0xd1ad, 0x80, 0 },
+	{ 0xd1ae, 0x00, 0 },
+	{ 0xd1af, 0x70, 0 },
+	{ 0xd1b0, 0xa8, 0 },
+	{ 0xd1b1, 0xe8, 0 },
+	{ 0xd1b2, 0x38, 0 },
+	{ 0xd1b3, 0x43, 0 },
+	{ 0xd1b4, 0xd8, 0 },
+	{ 0xd1b5, 0x03, 0 },
+	{ 0xd1b6, 0x20, 0 },
 	{ 0xd1b7, 0x00, 0 },
-	{ 0xd1b8, 0x9c, 0},
-	{ 0xd1b9, 0xa0, 0},
-	{ 0xd1ba, 0x00, 0},
-	{ 0xd1bb, 0x00, 0},
-	{ 0xd1bc, 0xa8, 0},
-	{ 0xd1bd, 0xc8, 0},
-	{ 0xd1be, 0x38, 0},
-	{ 0xd1bf, 0x42, 0},
+	{ 0xd1b8, 0x9c, 0 },
+	{ 0xd1b9, 0xa0, 0 },
+	{ 0xd1ba, 0x00, 0 },
+	{ 0xd1bb, 0x00, 0 },
+	{ 0xd1bc, 0xa8, 0 },
+	{ 0xd1bd, 0xc8, 0 },
+	{ 0xd1be, 0x38, 0 },
+	{ 0xd1bf, 0x42, 0 },
 	{ 0xd1c0, 0x8c, 0 },
-	{ 0xd1c1, 0x66, 0},
+	{ 0xd1c1, 0x66, 0 },
 	{ 0xd1c2, 0x00, 0 },
 	{ 0xd1c3, 0x00, 0 },
-	{ 0xd1c4, 0x9c, 0},
-	{ 0xd1c5, 0xa5, 0},
+	{ 0xd1c4, 0x9c, 0 },
+	{ 0xd1c5, 0xa5, 0 },
 	{ 0xd1c6, 0x00, 0 },
-	{ 0xd1c7, 0x01, 0},
-	{ 0xd1c8, 0xb8, 0},
-	{ 0xd1c9, 0x83, 0},
+	{ 0xd1c7, 0x01, 0 },
+	{ 0xd1c8, 0xb8, 0 },
+	{ 0xd1c9, 0x83, 0 },
 	{ 0xd1ca, 0x00, 0 },
-	{ 0xd1cb, 0x08, 0},
-	{ 0xd1cc, 0xa4, 0},
-	{ 0xd1cd, 0xa5, 0},
-	{ 0xd1ce, 0x00, 0},
-	{ 0xd1cf, 0xff, 0},
-	{ 0xd1d0, 0x8c, 0},
-	{ 0xd1d1, 0x67, 0},
-	{ 0xd1d2, 0x00, 0},
-	{ 0xd1d3, 0x00, 0},
-	{ 0xd1d4, 0xe0, 0},
-	{ 0xd1d5, 0x63, 0},
-	{ 0xd1d6, 0x20, 0},
-	{ 0xd1d7, 0x00, 0},
-	{ 0xd1d8, 0xa4, 0},
+	{ 0xd1cb, 0x08, 0 },
+	{ 0xd1cc, 0xa4, 0 },
+	{ 0xd1cd, 0xa5, 0 },
+	{ 0xd1ce, 0x00, 0 },
+	{ 0xd1cf, 0xff, 0 },
+	{ 0xd1d0, 0x8c, 0 },
+	{ 0xd1d1, 0x67, 0 },
+	{ 0xd1d2, 0x00, 0 },
+	{ 0xd1d3, 0x00, 0 },
+	{ 0xd1d4, 0xe0, 0 },
+	{ 0xd1d5, 0x63, 0 },
+	{ 0xd1d6, 0x20, 0 },
+	{ 0xd1d7, 0x00, 0 },
+	{ 0xd1d8, 0xa4, 0 },
 	{ 0xd1d9, 0x63, 0 },
-	{ 0xd1da, 0xff, 0},
-	{ 0xd1db, 0xff, 0},
-	{ 0xd1dc, 0xbc, 0},
-	{ 0xd1dd, 0x43, 0},
-	{ 0xd1de, 0x00, 0},
-	{ 0xd1df, 0x07, 0},
-	{ 0xd1e0, 0x0c, 0},
-	{ 0xd1e1, 0x00, 0},
-	{ 0xd1e2, 0x00, 0},
-	{ 0xd1e3, 0x5b, 0},
-	{ 0xd1e4, 0xbc, 0},
+	{ 0xd1da, 0xff, 0 },
+	{ 0xd1db, 0xff, 0 },
+	{ 0xd1dc, 0xbc, 0 },
+	{ 0xd1dd, 0x43, 0 },
+	{ 0xd1de, 0x00, 0 },
+	{ 0xd1df, 0x07, 0 },
+	{ 0xd1e0, 0x0c, 0 },
+	{ 0xd1e1, 0x00, 0 },
+	{ 0xd1e2, 0x00, 0 },
+	{ 0xd1e3, 0x5b, 0 },
+	{ 0xd1e4, 0xbc, 0 },
 	{ 0xd1e5, 0x05, 0 },
-	{ 0xd1e6, 0x00, 0},
-	{ 0xd1e7, 0x02, 0},
-	{ 0xd1e8, 0x03, 0},
-	{ 0xd1e9, 0xff, 0},
-	{ 0xd1ea, 0xff, 0},
-	{ 0xd1eb, 0xf6, 0},
-	{ 0xd1ec, 0x9c, 0},
-	{ 0xd1ed, 0xa0, 0},
-	{ 0xd1ee, 0x00, 0},
-	{ 0xd1ef, 0x00, 0},
-	{ 0xd1f0, 0xa8, 0},
-	{ 0xd1f1, 0xa4, 0},
-	{ 0xd1f2, 0x55, 0},
-	{ 0xd1f3, 0x86, 0},
-	{ 0xd1f4, 0x8c, 0},
-	{ 0xd1f5, 0x63, 0},
-	{ 0xd1f6, 0x00, 0},
-	{ 0xd1f7, 0x00, 0},
-	{ 0xd1f8, 0xa8, 0},
-	{ 0xd1f9, 0xc4, 0},
-	{ 0xd1fa, 0x6e, 0},
-	{ 0xd1fb, 0x45, 0},
-	{ 0xd1fc, 0xa8, 0},
-	{ 0xd1fd, 0xe4, 0},
-	{ 0xd1fe, 0x55, 0},
-	{ 0xd1ff, 0x87, 0},
-	{ 0xd200, 0xd8, 0},
-	{ 0xd201, 0x05, 0},
-	{ 0xd202, 0x18, 0},
-	{ 0xd203, 0x00, 0},
-	{ 0xd204, 0x8c, 0},
-	{ 0xd205, 0x66, 0},
-	{ 0xd206, 0x00, 0},
-	{ 0xd207, 0x00, 0},
+	{ 0xd1e6, 0x00, 0 },
+	{ 0xd1e7, 0x02, 0 },
+	{ 0xd1e8, 0x03, 0 },
+	{ 0xd1e9, 0xff, 0 },
+	{ 0xd1ea, 0xff, 0 },
+	{ 0xd1eb, 0xf6, 0 },
+	{ 0xd1ec, 0x9c, 0 },
+	{ 0xd1ed, 0xa0, 0 },
+	{ 0xd1ee, 0x00, 0 },
+	{ 0xd1ef, 0x00, 0 },
+	{ 0xd1f0, 0xa8, 0 },
+	{ 0xd1f1, 0xa4, 0 },
+	{ 0xd1f2, 0x55, 0 },
+	{ 0xd1f3, 0x86, 0 },
+	{ 0xd1f4, 0x8c, 0 },
+	{ 0xd1f5, 0x63, 0 },
+	{ 0xd1f6, 0x00, 0 },
+	{ 0xd1f7, 0x00, 0 },
+	{ 0xd1f8, 0xa8, 0 },
+	{ 0xd1f9, 0xc4, 0 },
+	{ 0xd1fa, 0x6e, 0 },
+	{ 0xd1fb, 0x45, 0 },
+	{ 0xd1fc, 0xa8, 0 },
+	{ 0xd1fd, 0xe4, 0 },
+	{ 0xd1fe, 0x55, 0 },
+	{ 0xd1ff, 0x87, 0 },
+	{ 0xd200, 0xd8, 0 },
+	{ 0xd201, 0x05, 0 },
+	{ 0xd202, 0x18, 0 },
+	{ 0xd203, 0x00, 0 },
+	{ 0xd204, 0x8c, 0 },
+	{ 0xd205, 0x66, 0 },
+	{ 0xd206, 0x00, 0 },
+	{ 0xd207, 0x00, 0 },
 	{ 0xd208, 0xa8, 0 },
-	{ 0xd209, 0xa4, 0},
+	{ 0xd209, 0xa4, 0 },
 	{ 0xd20a, 0x6e, 0 },
-	{ 0xd20b, 0x46, 0},
-	{ 0xd20c, 0xd8, 0},
-	{ 0xd20d, 0x07, 0},
-	{ 0xd20e, 0x18, 0},
-	{ 0xd20f, 0x00, 0},
-	{ 0xd210, 0xa8, 0},
+	{ 0xd20b, 0x46, 0 },
+	{ 0xd20c, 0xd8, 0 },
+	{ 0xd20d, 0x07, 0 },
+	{ 0xd20e, 0x18, 0 },
+	{ 0xd20f, 0x00, 0 },
+	{ 0xd210, 0xa8, 0 },
 	{ 0xd211, 0x84, 0 },
-	{ 0xd212, 0x55, 0},
-	{ 0xd213, 0x88, 0},
-	{ 0xd214, 0x8c, 0},
-	{ 0xd215, 0x65, 0},
-	{ 0xd216, 0x00, 0},
-	{ 0xd217, 0x00, 0},
-	{ 0xd218, 0xd8, 0},
-	{ 0xd219, 0x04, 0},
-	{ 0xd21a, 0x18, 0},
-	{ 0xd21b, 0x00, 0},
-	{ 0xd21c, 0x03, 0},
-	{ 0xd21d, 0xff, 0},
-	{ 0xd21e, 0xff, 0},
-	{ 0xd21f, 0xce, 0},
-	{ 0xd220, 0x19, 0},
-	{ 0xd221, 0x00, 0},
-	{ 0xd222, 0x80, 0},
-	{ 0xd223, 0x06, 0},
-	{ 0xd224, 0x8c, 0},
-	{ 0xd225, 0x63, 0},
-	{ 0xd226, 0x00, 0},
+	{ 0xd212, 0x55, 0 },
+	{ 0xd213, 0x88, 0 },
+	{ 0xd214, 0x8c, 0 },
+	{ 0xd215, 0x65, 0 },
+	{ 0xd216, 0x00, 0 },
+	{ 0xd217, 0x00, 0 },
+	{ 0xd218, 0xd8, 0 },
+	{ 0xd219, 0x04, 0 },
+	{ 0xd21a, 0x18, 0 },
+	{ 0xd21b, 0x00, 0 },
+	{ 0xd21c, 0x03, 0 },
+	{ 0xd21d, 0xff, 0 },
+	{ 0xd21e, 0xff, 0 },
+	{ 0xd21f, 0xce, 0 },
+	{ 0xd220, 0x19, 0 },
+	{ 0xd221, 0x00, 0 },
+	{ 0xd222, 0x80, 0 },
+	{ 0xd223, 0x06, 0 },
+	{ 0xd224, 0x8c, 0 },
+	{ 0xd225, 0x63, 0 },
+	{ 0xd226, 0x00, 0 },
 	{ 0xd227, 0x00, 0 },
-	{ 0xd228, 0xa4, 0},
-	{ 0xd229, 0x63, 0},
-	{ 0xd22a, 0x00, 0},
-	{ 0xd22b, 0x40, 0},
-	{ 0xd22c, 0xbc, 0},
-	{ 0xd22d, 0x23, 0},
+	{ 0xd228, 0xa4, 0 },
+	{ 0xd229, 0x63, 0 },
+	{ 0xd22a, 0x00, 0 },
+	{ 0xd22b, 0x40, 0 },
+	{ 0xd22c, 0xbc, 0 },
+	{ 0xd22d, 0x23, 0 },
 	{ 0xd22e, 0x00, 0 },
 	{ 0xd22f, 0x00, 0 },
-	{ 0xd230, 0x13, 0},
-	{ 0xd231, 0xff, 0},
+	{ 0xd230, 0x13, 0 },
+	{ 0xd231, 0xff, 0 },
 	{ 0xd232, 0xff, 0 },
-	{ 0xd233, 0xc8, 0},
-	{ 0xd234, 0x9d, 0},
-	{ 0xd235, 0x00, 0},
-	{ 0xd236, 0x00, 0},
-	{ 0xd237, 0x40, 0},
-	{ 0xd238, 0xa8, 0},
-	{ 0xd239, 0x64, 0},
-	{ 0xd23a, 0x55, 0},
-	{ 0xd23b, 0x86, 0},
+	{ 0xd233, 0xc8, 0 },
+	{ 0xd234, 0x9d, 0 },
+	{ 0xd235, 0x00, 0 },
+	{ 0xd236, 0x00, 0 },
+	{ 0xd237, 0x40, 0 },
+	{ 0xd238, 0xa8, 0 },
+	{ 0xd239, 0x64, 0 },
+	{ 0xd23a, 0x55, 0 },
+	{ 0xd23b, 0x86, 0 },
 	{ 0xd23c, 0xa8, 0 },
-	{ 0xd23d, 0xa4, 0},
-	{ 0xd23e, 0x55, 0},
-	{ 0xd23f, 0x87, 0},
-	{ 0xd240, 0xd8, 0},
-	{ 0xd241, 0x03, 0},
-	{ 0xd242, 0x40, 0},
-	{ 0xd243, 0x00, 0},
-	{ 0xd244, 0xa8, 0},
-	{ 0xd245, 0x64, 0},
-	{ 0xd246, 0x55, 0},
-	{ 0xd247, 0x88, 0},
-	{ 0xd248, 0xd8, 0},
-	{ 0xd249, 0x05, 0},
-	{ 0xd24a, 0x40, 0},
-	{ 0xd24b, 0x00, 0},
-	{ 0xd24c, 0xd8, 0},
-	{ 0xd24d, 0x03, 0},
-	{ 0xd24e, 0x40, 0},
+	{ 0xd23d, 0xa4, 0 },
+	{ 0xd23e, 0x55, 0 },
+	{ 0xd23f, 0x87, 0 },
+	{ 0xd240, 0xd8, 0 },
+	{ 0xd241, 0x03, 0 },
+	{ 0xd242, 0x40, 0 },
+	{ 0xd243, 0x00, 0 },
+	{ 0xd244, 0xa8, 0 },
+	{ 0xd245, 0x64, 0 },
+	{ 0xd246, 0x55, 0 },
+	{ 0xd247, 0x88, 0 },
+	{ 0xd248, 0xd8, 0 },
+	{ 0xd249, 0x05, 0 },
+	{ 0xd24a, 0x40, 0 },
+	{ 0xd24b, 0x00, 0 },
+	{ 0xd24c, 0xd8, 0 },
+	{ 0xd24d, 0x03, 0 },
+	{ 0xd24e, 0x40, 0 },
 	{ 0xd24f, 0x00, 0 },
-	{ 0xd250, 0x03, 0},
+	{ 0xd250, 0x03, 0 },
 	{ 0xd251, 0xff, 0 },
-	{ 0xd252, 0xff, 0},
-	{ 0xd253, 0xc1, 0},
-	{ 0xd254, 0x19, 0},
+	{ 0xd252, 0xff, 0 },
+	{ 0xd253, 0xc1, 0 },
+	{ 0xd254, 0x19, 0 },
 	{ 0xd255, 0x00, 0 },
-	{ 0xd256, 0x80, 0},
-	{ 0xd257, 0x06, 0},
-	{ 0xd258, 0x94, 0},
-	{ 0xd259, 0x84, 0},
-	{ 0xd25a, 0x00, 0},
-	{ 0xd25b, 0x72, 0},
-	{ 0xd25c, 0xe5, 0},
-	{ 0xd25d, 0xa4, 0},
-	{ 0xd25e, 0x60, 0},
-	{ 0xd25f, 0x00, 0},
-	{ 0xd260, 0x0c, 0},
-	{ 0xd261, 0x00, 0},
+	{ 0xd256, 0x80, 0 },
+	{ 0xd257, 0x06, 0 },
+	{ 0xd258, 0x94, 0 },
+	{ 0xd259, 0x84, 0 },
+	{ 0xd25a, 0x00, 0 },
+	{ 0xd25b, 0x72, 0 },
+	{ 0xd25c, 0xe5, 0 },
+	{ 0xd25d, 0xa4, 0 },
+	{ 0xd25e, 0x60, 0 },
+	{ 0xd25f, 0x00, 0 },
+	{ 0xd260, 0x0c, 0 },
+	{ 0xd261, 0x00, 0 },
 	{ 0xd262, 0x00, 0 },
-	{ 0xd263, 0x3f, 0},
-	{ 0xd264, 0x9d, 0},
-	{ 0xd265, 0x60, 0},
-	{ 0xd266, 0x01, 0},
-	{ 0xd267, 0x00, 0},
-	{ 0xd268, 0x85, 0},
-	{ 0xd269, 0x4e, 0},
+	{ 0xd263, 0x3f, 0 },
+	{ 0xd264, 0x9d, 0 },
+	{ 0xd265, 0x60, 0 },
+	{ 0xd266, 0x01, 0 },
+	{ 0xd267, 0x00, 0 },
+	{ 0xd268, 0x85, 0 },
+	{ 0xd269, 0x4e, 0 },
 	{ 0xd26a, 0x00, 0 },
-	{ 0xd26b, 0x00, 0},
-	{ 0xd26c, 0x98, 0},
-	{ 0xd26d, 0x70, 0},
+	{ 0xd26b, 0x00, 0 },
+	{ 0xd26c, 0x98, 0 },
+	{ 0xd26d, 0x70, 0 },
 	{ 0xd26e, 0x00, 0 },
 	{ 0xd26f, 0x00, 0 },
-	{ 0xd270, 0x8c, 0},
-	{ 0xd271, 0x8a, 0},
+	{ 0xd270, 0x8c, 0 },
+	{ 0xd271, 0x8a, 0 },
 	{ 0xd272, 0x00, 0 },
-	{ 0xd273, 0x6f, 0},
-	{ 0xd274, 0xe5, 0},
-	{ 0xd275, 0x63, 0},
-	{ 0xd276, 0x20, 0},
-	{ 0xd277, 0x00, 0},
-	{ 0xd278, 0x10, 0},
-	{ 0xd279, 0x00, 0},
-	{ 0xd27a, 0x00, 0},
-	{ 0xd27b, 0x07, 0},
+	{ 0xd273, 0x6f, 0 },
+	{ 0xd274, 0xe5, 0 },
+	{ 0xd275, 0x63, 0 },
+	{ 0xd276, 0x20, 0 },
+	{ 0xd277, 0x00, 0 },
+	{ 0xd278, 0x10, 0 },
+	{ 0xd279, 0x00, 0 },
+	{ 0xd27a, 0x00, 0 },
+	{ 0xd27b, 0x07, 0 },
 	{ 0xd27c, 0x15, 0 },
 	{ 0xd27d, 0x00, 0 },
 	{ 0xd27e, 0x00, 0 },
 	{ 0xd27f, 0x00, 0 },
-	{ 0xd280, 0x8c, 0},
-	{ 0xd281, 0xaa, 0},
-	{ 0xd282, 0x00, 0},
-	{ 0xd283, 0x6e, 0},
-	{ 0xd284, 0xe0, 0},
+	{ 0xd280, 0x8c, 0 },
+	{ 0xd281, 0xaa, 0 },
+	{ 0xd282, 0x00, 0 },
+	{ 0xd283, 0x6e, 0 },
+	{ 0xd284, 0xe0, 0 },
 	{ 0xd285, 0x63, 0 },
-	{ 0xd286, 0x28, 0},
-	{ 0xd287, 0x02, 0},
-	{ 0xd288, 0xe0, 0},
-	{ 0xd289, 0x84, 0},
-	{ 0xd28a, 0x28, 0},
-	{ 0xd28b, 0x02, 0},
-	{ 0xd28c, 0x07, 0},
-	{ 0xd28d, 0xff, 0},
-	{ 0xd28e, 0xf8, 0},
-	{ 0xd28f, 0x66, 0},
-	{ 0xd290, 0xe0, 0},
-	{ 0xd291, 0x63, 0},
-	{ 0xd292, 0x5b, 0},
-	{ 0xd293, 0x06, 0},
-	{ 0xd294, 0x8c, 0},
-	{ 0xd295, 0x6a, 0},
+	{ 0xd286, 0x28, 0 },
+	{ 0xd287, 0x02, 0 },
+	{ 0xd288, 0xe0, 0 },
+	{ 0xd289, 0x84, 0 },
+	{ 0xd28a, 0x28, 0 },
+	{ 0xd28b, 0x02, 0 },
+	{ 0xd28c, 0x07, 0 },
+	{ 0xd28d, 0xff, 0 },
+	{ 0xd28e, 0xf8, 0 },
+	{ 0xd28f, 0x66, 0 },
+	{ 0xd290, 0xe0, 0 },
+	{ 0xd291, 0x63, 0 },
+	{ 0xd292, 0x5b, 0 },
+	{ 0xd293, 0x06, 0 },
+	{ 0xd294, 0x8c, 0 },
+	{ 0xd295, 0x6a, 0 },
 	{ 0xd296, 0x00, 0 },
-	{ 0xd297, 0x77, 0},
-	{ 0xd298, 0xe0, 0},
-	{ 0xd299, 0x63, 0},
-	{ 0xd29a, 0x5b, 0},
-	{ 0xd29b, 0x06, 0},
-	{ 0xd29c, 0xbd, 0},
-	{ 0xd29d, 0x63, 0},
+	{ 0xd297, 0x77, 0 },
+	{ 0xd298, 0xe0, 0 },
+	{ 0xd299, 0x63, 0 },
+	{ 0xd29a, 0x5b, 0 },
+	{ 0xd29b, 0x06, 0 },
+	{ 0xd29c, 0xbd, 0 },
+	{ 0xd29d, 0x63, 0 },
 	{ 0xd29e, 0x00, 0 },
 	{ 0xd29f, 0x00, 0 },
-	{ 0xd2a0, 0x0c, 0},
-	{ 0xd2a1, 0x00, 0},
-	{ 0xd2a2, 0x00, 0},
-	{ 0xd2a3, 0x3c, 0},
-	{ 0xd2a4, 0x15, 0},
-	{ 0xd2a5, 0x00, 0},
-	{ 0xd2a6, 0x00, 0},
+	{ 0xd2a0, 0x0c, 0 },
+	{ 0xd2a1, 0x00, 0 },
+	{ 0xd2a2, 0x00, 0 },
+	{ 0xd2a3, 0x3c, 0 },
+	{ 0xd2a4, 0x15, 0 },
+	{ 0xd2a5, 0x00, 0 },
+	{ 0xd2a6, 0x00, 0 },
 	{ 0xd2a7, 0x00, 0 },
-	{ 0xd2a8, 0x8c, 0},
-	{ 0xd2a9, 0x8a, 0},
-	{ 0xd2aa, 0x00, 0},
-	{ 0xd2ab, 0x78, 0},
-	{ 0xd2ac, 0xb8, 0},
-	{ 0xd2ad, 0x63, 0},
+	{ 0xd2a8, 0x8c, 0 },
+	{ 0xd2a9, 0x8a, 0 },
+	{ 0xd2aa, 0x00, 0 },
+	{ 0xd2ab, 0x78, 0 },
+	{ 0xd2ac, 0xb8, 0 },
+	{ 0xd2ad, 0x63, 0 },
 	{ 0xd2ae, 0x00, 0 },
-	{ 0xd2af, 0x88, 0},
-	{ 0xd2b0, 0xe1, 0},
-	{ 0xd2b1, 0x64, 0},
-	{ 0xd2b2, 0x5b, 0},
-	{ 0xd2b3, 0x06, 0},
-	{ 0xd2b4, 0xbd, 0},
-	{ 0xd2b5, 0x6b, 0},
-	{ 0xd2b6, 0x00, 0},
+	{ 0xd2af, 0x88, 0 },
+	{ 0xd2b0, 0xe1, 0 },
+	{ 0xd2b1, 0x64, 0 },
+	{ 0xd2b2, 0x5b, 0 },
+	{ 0xd2b3, 0x06, 0 },
+	{ 0xd2b4, 0xbd, 0 },
+	{ 0xd2b5, 0x6b, 0 },
+	{ 0xd2b6, 0x00, 0 },
 	{ 0xd2b7, 0x00, 0 },
-	{ 0xd2b8, 0x0c, 0},
-	{ 0xd2b9, 0x00, 0},
+	{ 0xd2b8, 0x0c, 0 },
+	{ 0xd2b9, 0x00, 0 },
 	{ 0xd2ba, 0x00, 0 },
-	{ 0xd2bb, 0x34, 0},
-	{ 0xd2bc, 0xd4, 0},
-	{ 0xd2bd, 0x01, 0},
+	{ 0xd2bb, 0x34, 0 },
+	{ 0xd2bc, 0xd4, 0 },
+	{ 0xd2bd, 0x01, 0 },
 	{ 0xd2be, 0x18, 0 },
-	{ 0xd2bf, 0x14, 0},
-	{ 0xd2c0, 0xb9, 0},
-	{ 0xd2c1, 0x6b, 0},
+	{ 0xd2bf, 0x14, 0 },
+	{ 0xd2c0, 0xb9, 0 },
+	{ 0xd2c1, 0x6b, 0 },
 	{ 0xd2c2, 0x00, 0 },
-	{ 0xd2c3, 0x88, 0},
-	{ 0xd2c4, 0x85, 0},
-	{ 0xd2c5, 0x01, 0},
-	{ 0xd2c6, 0x00, 0},
-	{ 0xd2c7, 0x14, 0},
-	{ 0xd2c8, 0xbd, 0},
-	{ 0xd2c9, 0x68, 0},
+	{ 0xd2c3, 0x88, 0 },
+	{ 0xd2c4, 0x85, 0 },
+	{ 0xd2c5, 0x01, 0 },
+	{ 0xd2c6, 0x00, 0 },
+	{ 0xd2c7, 0x14, 0 },
+	{ 0xd2c8, 0xbd, 0 },
+	{ 0xd2c9, 0x68, 0 },
 	{ 0xd2ca, 0x00, 0 },
-	{ 0xd2cb, 0x00, 0},
-	{ 0xd2cc, 0x0c, 0},
-	{ 0xd2cd, 0x00, 0},
+	{ 0xd2cb, 0x00, 0 },
+	{ 0xd2cc, 0x0c, 0 },
+	{ 0xd2cd, 0x00, 0 },
 	{ 0xd2ce, 0x00, 0 },
-	{ 0xd2cf, 0x2c, 0},
-	{ 0xd2d0, 0xd4, 0},
-	{ 0xd2d1, 0x01, 0},
-	{ 0xd2d2, 0x58, 0},
-	{ 0xd2d3, 0x18, 0},
-	{ 0xd2d4, 0x84, 0},
-	{ 0xd2d5, 0x81, 0},
+	{ 0xd2cf, 0x2c, 0 },
+	{ 0xd2d0, 0xd4, 0 },
+	{ 0xd2d1, 0x01, 0 },
+	{ 0xd2d2, 0x58, 0 },
+	{ 0xd2d3, 0x18, 0 },
+	{ 0xd2d4, 0x84, 0 },
+	{ 0xd2d5, 0x81, 0 },
 	{ 0xd2d6, 0x00, 0 },
-	{ 0xd2d7, 0x14, 0},
-	{ 0xd2d8, 0xbd, 0},
-	{ 0xd2d9, 0xa4, 0},
-	{ 0xd2da, 0x01, 0},
-	{ 0xd2db, 0x00, 0},
-	{ 0xd2dc, 0x10, 0},
-	{ 0xd2dd, 0x00, 0},
-	{ 0xd2de, 0x00, 0},
-	{ 0xd2df, 0x05, 0},
-	{ 0xd2e0, 0x84, 0},
-	{ 0xd2e1, 0xc1, 0},
-	{ 0xd2e2, 0x00, 0},
-	{ 0xd2e3, 0x18, 0},
-	{ 0xd2e4, 0x9c, 0},
-	{ 0xd2e5, 0xa0, 0},
-	{ 0xd2e6, 0x01, 0},
+	{ 0xd2d7, 0x14, 0 },
+	{ 0xd2d8, 0xbd, 0 },
+	{ 0xd2d9, 0xa4, 0 },
+	{ 0xd2da, 0x01, 0 },
+	{ 0xd2db, 0x00, 0 },
+	{ 0xd2dc, 0x10, 0 },
+	{ 0xd2dd, 0x00, 0 },
+	{ 0xd2de, 0x00, 0 },
+	{ 0xd2df, 0x05, 0 },
+	{ 0xd2e0, 0x84, 0 },
+	{ 0xd2e1, 0xc1, 0 },
+	{ 0xd2e2, 0x00, 0 },
+	{ 0xd2e3, 0x18, 0 },
+	{ 0xd2e4, 0x9c, 0 },
+	{ 0xd2e5, 0xa0, 0 },
+	{ 0xd2e6, 0x01, 0 },
 	{ 0xd2e7, 0x00, 0 },
-	{ 0xd2e8, 0xd4, 0},
-	{ 0xd2e9, 0x01, 0},
-	{ 0xd2ea, 0x28, 0},
-	{ 0xd2eb, 0x14, 0},
-	{ 0xd2ec, 0x84, 0},
-	{ 0xd2ed, 0xc1, 0},
+	{ 0xd2e8, 0xd4, 0 },
+	{ 0xd2e9, 0x01, 0 },
+	{ 0xd2ea, 0x28, 0 },
+	{ 0xd2eb, 0x14, 0 },
+	{ 0xd2ec, 0x84, 0 },
+	{ 0xd2ed, 0xc1, 0 },
 	{ 0xd2ee, 0x00, 0 },
-	{ 0xd2ef, 0x18, 0},
-	{ 0xd2f0, 0xbd, 0},
-	{ 0xd2f1, 0x66, 0},
+	{ 0xd2ef, 0x18, 0 },
+	{ 0xd2f0, 0xbd, 0 },
+	{ 0xd2f1, 0x66, 0 },
 	{ 0xd2f2, 0x00, 0 },
-	{ 0xd2f3, 0x00, 0},
-	{ 0xd2f4, 0x0c, 0},
-	{ 0xd2f5, 0x00, 0},
-	{ 0xd2f6, 0x00, 0},
-	{ 0xd2f7, 0x20, 0},
-	{ 0xd2f8, 0x9d, 0},
-	{ 0xd2f9, 0x00, 0},
+	{ 0xd2f3, 0x00, 0 },
+	{ 0xd2f4, 0x0c, 0 },
+	{ 0xd2f5, 0x00, 0 },
+	{ 0xd2f6, 0x00, 0 },
+	{ 0xd2f7, 0x20, 0 },
+	{ 0xd2f8, 0x9d, 0 },
+	{ 0xd2f9, 0x00, 0 },
 	{ 0xd2fa, 0x00, 0 },
-	{ 0xd2fb, 0x00, 0},
-	{ 0xd2fc, 0x84, 0},
-	{ 0xd2fd, 0x61, 0},
+	{ 0xd2fb, 0x00, 0 },
+	{ 0xd2fc, 0x84, 0 },
+	{ 0xd2fd, 0x61, 0 },
 	{ 0xd2fe, 0x00, 0 },
-	{ 0xd2ff, 0x18, 0},
-	{ 0xd300, 0xbd, 0},
-	{ 0xd301, 0xa3, 0},
-	{ 0xd302, 0x01, 0},
-	{ 0xd303, 0x00, 0},
-	{ 0xd304, 0x10, 0},
-	{ 0xd305, 0x00, 0},
-	{ 0xd306, 0x00, 0},
-	{ 0xd307, 0x03, 0},
+	{ 0xd2ff, 0x18, 0 },
+	{ 0xd300, 0xbd, 0 },
+	{ 0xd301, 0xa3, 0 },
+	{ 0xd302, 0x01, 0 },
+	{ 0xd303, 0x00, 0 },
+	{ 0xd304, 0x10, 0 },
+	{ 0xd305, 0x00, 0 },
+	{ 0xd306, 0x00, 0 },
+	{ 0xd307, 0x03, 0 },
 	{ 0xd308, 0x9c, 0 },
-	{ 0xd309, 0x80, 0},
-	{ 0xd30a, 0x01, 0},
-	{ 0xd30b, 0x00, 0},
-	{ 0xd30c, 0xd4, 0},
-	{ 0xd30d, 0x01, 0},
-	{ 0xd30e, 0x20, 0},
-	{ 0xd30f, 0x18, 0},
-	{ 0xd310, 0x18, 0},
-	{ 0xd311, 0x60, 0},
-	{ 0xd312, 0x80, 0},
-	{ 0xd313, 0x06, 0},
-	{ 0xd314, 0x85, 0},
-	{ 0xd315, 0x01, 0},
+	{ 0xd309, 0x80, 0 },
+	{ 0xd30a, 0x01, 0 },
+	{ 0xd30b, 0x00, 0 },
+	{ 0xd30c, 0xd4, 0 },
+	{ 0xd30d, 0x01, 0 },
+	{ 0xd30e, 0x20, 0 },
+	{ 0xd30f, 0x18, 0 },
+	{ 0xd310, 0x18, 0 },
+	{ 0xd311, 0x60, 0 },
+	{ 0xd312, 0x80, 0 },
+	{ 0xd313, 0x06, 0 },
+	{ 0xd314, 0x85, 0 },
+	{ 0xd315, 0x01, 0 },
 	{ 0xd316, 0x00, 0 },
-	{ 0xd317, 0x14, 0},
-	{ 0xd318, 0xa8, 0},
-	{ 0xd319, 0x83, 0},
-	{ 0xd31a, 0x38, 0},
-	{ 0xd31b, 0x29, 0},
-	{ 0xd31c, 0xa8, 0},
-	{ 0xd31d, 0xc3, 0},
-	{ 0xd31e, 0x40, 0},
-	{ 0xd31f, 0x08, 0},
-	{ 0xd320, 0x8c, 0},
-	{ 0xd321, 0x84, 0},
-	{ 0xd322, 0x00, 0},
-	{ 0xd323, 0x00, 0},
-	{ 0xd324, 0xa8, 0},
-	{ 0xd325, 0xa3, 0},
-	{ 0xd326, 0x38, 0},
-	{ 0xd327, 0x2a, 0},
-	{ 0xd328, 0xa8, 0},
-	{ 0xd329, 0xe3, 0},
-	{ 0xd32a, 0x40, 0},
-	{ 0xd32b, 0x09, 0},
-	{ 0xd32c, 0xe0, 0},
-	{ 0xd32d, 0x64, 0},
-	{ 0xd32e, 0x40, 0},
-	{ 0xd32f, 0x00, 0},
+	{ 0xd317, 0x14, 0 },
+	{ 0xd318, 0xa8, 0 },
+	{ 0xd319, 0x83, 0 },
+	{ 0xd31a, 0x38, 0 },
+	{ 0xd31b, 0x29, 0 },
+	{ 0xd31c, 0xa8, 0 },
+	{ 0xd31d, 0xc3, 0 },
+	{ 0xd31e, 0x40, 0 },
+	{ 0xd31f, 0x08, 0 },
+	{ 0xd320, 0x8c, 0 },
+	{ 0xd321, 0x84, 0 },
+	{ 0xd322, 0x00, 0 },
+	{ 0xd323, 0x00, 0 },
+	{ 0xd324, 0xa8, 0 },
+	{ 0xd325, 0xa3, 0 },
+	{ 0xd326, 0x38, 0 },
+	{ 0xd327, 0x2a, 0 },
+	{ 0xd328, 0xa8, 0 },
+	{ 0xd329, 0xe3, 0 },
+	{ 0xd32a, 0x40, 0 },
+	{ 0xd32b, 0x09, 0 },
+	{ 0xd32c, 0xe0, 0 },
+	{ 0xd32d, 0x64, 0 },
+	{ 0xd32e, 0x40, 0 },
+	{ 0xd32f, 0x00, 0 },
 	{ 0xd330, 0xd8, 0 },
-	{ 0xd331, 0x06, 0},
-	{ 0xd332, 0x18, 0},
+	{ 0xd331, 0x06, 0 },
+	{ 0xd332, 0x18, 0 },
 	{ 0xd333, 0x00, 0 },
-	{ 0xd334, 0x8c, 0},
-	{ 0xd335, 0x65, 0},
-	{ 0xd336, 0x00, 0},
+	{ 0xd334, 0x8c, 0 },
+	{ 0xd335, 0x65, 0 },
+	{ 0xd336, 0x00, 0 },
 	{ 0xd337, 0x00, 0 },
-	{ 0xd338, 0x84, 0},
-	{ 0xd339, 0x81, 0},
+	{ 0xd338, 0x84, 0 },
+	{ 0xd339, 0x81, 0 },
 	{ 0xd33a, 0x00, 0 },
-	{ 0xd33b, 0x18, 0},
-	{ 0xd33c, 0xe3, 0},
-	{ 0xd33d, 0xe3, 0},
-	{ 0xd33e, 0x20, 0},
-	{ 0xd33f, 0x00, 0},
-	{ 0xd340, 0xd8, 0},
-	{ 0xd341, 0x07, 0},
-	{ 0xd342, 0xf8, 0},
-	{ 0xd343, 0x00, 0},
-	{ 0xd344, 0x03, 0},
-	{ 0xd345, 0xff, 0},
-	{ 0xd346, 0xff, 0},
-	{ 0xd347, 0x6f, 0},
-	{ 0xd348, 0x18, 0},
-	{ 0xd349, 0x60, 0},
+	{ 0xd33b, 0x18, 0 },
+	{ 0xd33c, 0xe3, 0 },
+	{ 0xd33d, 0xe3, 0 },
+	{ 0xd33e, 0x20, 0 },
+	{ 0xd33f, 0x00, 0 },
+	{ 0xd340, 0xd8, 0 },
+	{ 0xd341, 0x07, 0 },
+	{ 0xd342, 0xf8, 0 },
+	{ 0xd343, 0x00, 0 },
+	{ 0xd344, 0x03, 0 },
+	{ 0xd345, 0xff, 0 },
+	{ 0xd346, 0xff, 0 },
+	{ 0xd347, 0x6f, 0 },
+	{ 0xd348, 0x18, 0 },
+	{ 0xd349, 0x60, 0 },
 	{ 0xd34a, 0x00, 0 },
-	{ 0xd34b, 0x01, 0},
-	{ 0xd34c, 0x0f, 0},
+	{ 0xd34b, 0x01, 0 },
+	{ 0xd34c, 0xf , 0 },
 	{ 0xd34d, 0xff, 0 },
 	{ 0xd34e, 0xff, 0 },
-	{ 0xd34f, 0x9d, 0},
-	{ 0xd350, 0x18, 0},
-	{ 0xd351, 0x60, 0},
-	{ 0xd352, 0x80, 0},
-	{ 0xd353, 0x06, 0},
-	{ 0xd354, 0x00, 0},
+	{ 0xd34f, 0x9d, 0 },
+	{ 0xd350, 0x18, 0 },
+	{ 0xd351, 0x60, 0 },
+	{ 0xd352, 0x80, 0 },
+	{ 0xd353, 0x06, 0 },
+	{ 0xd354, 0x00, 0 },
 	{ 0xd355, 0x00, 0 },
-	{ 0xd356, 0x00, 0},
-	{ 0xd357, 0x11, 0},
-	{ 0xd358, 0xa8, 0},
-	{ 0xd359, 0x83, 0},
-	{ 0xd35a, 0x6e, 0},
-	{ 0xd35b, 0x43, 0},
-	{ 0xd35c, 0xe0, 0},
-	{ 0xd35d, 0x6c, 0},
-	{ 0xd35e, 0x28, 0},
-	{ 0xd35f, 0x02, 0},
-	{ 0xd360, 0xe0, 0},
-	{ 0xd361, 0x84, 0},
-	{ 0xd362, 0x28, 0},
-	{ 0xd363, 0x02, 0},
-	{ 0xd364, 0x07, 0},
-	{ 0xd365, 0xff, 0},
-	{ 0xd366, 0xf8, 0},
-	{ 0xd367, 0x30, 0},
-	{ 0xd368, 0xb8, 0},
-	{ 0xd369, 0x63, 0},
-	{ 0xd36a, 0x00, 0},
-	{ 0xd36b, 0x08, 0},
-	{ 0xd36c, 0x03, 0},
-	{ 0xd36d, 0xff, 0},
-	{ 0xd36e, 0xff, 0},
-	{ 0xd36f, 0xc0, 0},
-	{ 0xd370, 0x85, 0},
-	{ 0xd371, 0x4e, 0},
-	{ 0xd372, 0x00, 0},
-	{ 0xd373, 0x00, 0},
-	{ 0xd374, 0x03, 0},
-	{ 0xd375, 0xff, 0},
-	{ 0xd376, 0xff, 0},
-	{ 0xd377, 0xe7, 0},
-	{ 0xd378, 0xd4, 0},
-	{ 0xd379, 0x01, 0},
-	{ 0xd37a, 0x40, 0},
-	{ 0xd37b, 0x18, 0},
-	{ 0xd37c, 0x9c, 0},
-	{ 0xd37d, 0x60, 0},
-	{ 0xd37e, 0x00, 0},
-	{ 0xd37f, 0x00, 0},
-	{ 0xd380, 0x03, 0},
-	{ 0xd381, 0xff, 0},
-	{ 0xd382, 0xff, 0},
-	{ 0xd383, 0xdb, 0},
-	{ 0xd384, 0xd4, 0},
-	{ 0xd385, 0x01, 0},
-	{ 0xd386, 0x18, 0},
-	{ 0xd387, 0x14, 0},
-	{ 0xd388, 0x03, 0},
-	{ 0xd389, 0xff, 0},
-	{ 0xd38a, 0xff, 0},
-	{ 0xd38b, 0xce, 0},
-	{ 0xd38c, 0x9d, 0},
-	{ 0xd38d, 0x6b, 0},
-	{ 0xd38e, 0x00, 0},
-	{ 0xd38f, 0xff, 0},
-	{ 0xd390, 0x03, 0},
-	{ 0xd391, 0xff, 0},
-	{ 0xd392, 0xff, 0},
-	{ 0xd393, 0xc6, 0},
-	{ 0xd394, 0x9c, 0},
-	{ 0xd395, 0x63, 0},
-	{ 0xd396, 0x00, 0},
-	{ 0xd397, 0xff, 0},
-	{ 0xd398, 0xa8, 0},
-	{ 0xd399, 0xe3, 0},
-	{ 0xd39a, 0x38, 0},
-	{ 0xd39b, 0x0f, 0},
-	{ 0xd39c, 0x8c, 0},
-	{ 0xd39d, 0x84, 0},
-	{ 0xd39e, 0x00, 0},
-	{ 0xd39f, 0x00, 0},
-	{ 0xd3a0, 0xa8, 0},
-	{ 0xd3a1, 0xa3, 0},
-	{ 0xd3a2, 0x38, 0},
-	{ 0xd3a3, 0x0e, 0},
-	{ 0xd3a4, 0xa8, 0},
-	{ 0xd3a5, 0xc3, 0},
-	{ 0xd3a6, 0x6e, 0},
-	{ 0xd3a7, 0x42, 0},
-	{ 0xd3a8, 0xd8, 0},
-	{ 0xd3a9, 0x07, 0},
-	{ 0xd3aa, 0x20, 0},
-	{ 0xd3ab, 0x00, 0},
-	{ 0xd3ac, 0x8c, 0},
-	{ 0xd3ad, 0x66, 0},
-	{ 0xd3ae, 0x00, 0},
-	{ 0xd3af, 0x00, 0},
-	{ 0xd3b0, 0xd8, 0},
-	{ 0xd3b1, 0x05, 0},
-	{ 0xd3b2, 0x18, 0},
-	{ 0xd3b3, 0x00, 0},
-	{ 0xd3b4, 0x85, 0},
-	{ 0xd3b5, 0x21, 0},
-	{ 0xd3b6, 0x00, 0},
-	{ 0xd3b7, 0x00, 0},
-	{ 0xd3b8, 0x85, 0},
-	{ 0xd3b9, 0x41, 0},
-	{ 0xd3ba, 0x00, 0},
-	{ 0xd3bb, 0x04, 0},
-	{ 0xd3bc, 0x85, 0},
-	{ 0xd3bd, 0x81, 0},
-	{ 0xd3be, 0x00, 0},
-	{ 0xd3bf, 0x08, 0},
-	{ 0xd3c0, 0x85, 0},
-	{ 0xd3c1, 0xc1, 0},
-	{ 0xd3c2, 0x00, 0},
-	{ 0xd3c3, 0x0c, 0},
-	{ 0xd3c4, 0x86, 0},
-	{ 0xd3c5, 0x01, 0},
-	{ 0xd3c6, 0x00, 0},
-	{ 0xd3c7, 0x10, 0},
-	{ 0xd3c8, 0x44, 0},
-	{ 0xd3c9, 0x00, 0},
-	{ 0xd3ca, 0x48, 0},
-	{ 0xd3cb, 0x00, 0},
-	{ 0xd3cc, 0x9c, 0},
-	{ 0xd3cd, 0x21, 0},
-	{ 0xd3ce, 0x00, 0},
-	{ 0xd3cf, 0x1c, 0},
-	{ 0xd3d0, 0x9c, 0},
-	{ 0xd3d1, 0x21, 0},
-	{ 0xd3d2, 0xff, 0},
-	{ 0xd3d3, 0xfc, 0},
-	{ 0xd3d4, 0xd4, 0},
-	{ 0xd3d5, 0x01, 0},
-	{ 0xd3d6, 0x48, 0},
-	{ 0xd3d7, 0x00, 0},
-	{ 0xd3d8, 0x18, 0},
-	{ 0xd3d9, 0x60, 0},
-	{ 0xd3da, 0x00, 0},
-	{ 0xd3db, 0x01, 0},
-	{ 0xd3dc, 0xa8, 0},
-	{ 0xd3dd, 0x63, 0},
-	{ 0xd3de, 0x07, 0},
-	{ 0xd3df, 0x80, 0},
-	{ 0xd3e0, 0x8c, 0},
-	{ 0xd3e1, 0x63, 0},
-	{ 0xd3e2, 0x00, 0},
-	{ 0xd3e3, 0x68, 0},
-	{ 0xd3e4, 0xbc, 0},
-	{ 0xd3e5, 0x03, 0},
-	{ 0xd3e6, 0x00, 0},
-	{ 0xd3e7, 0x00, 0},
-	{ 0xd3e8, 0x10, 0},
-	{ 0xd3e9, 0x00, 0},
-	{ 0xd3ea, 0x00, 0},
-	{ 0xd3eb, 0x0c, 0},
-	{ 0xd3ec, 0x15, 0},
-	{ 0xd3ed, 0x00, 0},
-	{ 0xd3ee, 0x00, 0},
-	{ 0xd3ef, 0x00, 0},
-	{ 0xd3f0, 0x07, 0},
-	{ 0xd3f1, 0xff, 0},
-	{ 0xd3f2, 0xd9, 0},
-	{ 0xd3f3, 0x98, 0},
-	{ 0xd3f4, 0x15, 0},
-	{ 0xd3f5, 0x00, 0},
-	{ 0xd3f6, 0x00, 0},
-	{ 0xd3f7, 0x00, 0},
-	{ 0xd3f8, 0x18, 0},
-	{ 0xd3f9, 0x60, 0},
-	{ 0xd3fa, 0x80, 0},
-	{ 0xd3fb, 0x06, 0},
-	{ 0xd3fc, 0xa8, 0},
-	{ 0xd3fd, 0x63, 0},
-	{ 0xd3fe, 0xc4, 0},
-	{ 0xd3ff, 0xb8, 0},
-	{ 0xd400, 0x8c, 0},
-	{ 0xd401, 0x63, 0},
-	{ 0xd402, 0x00, 0},
-	{ 0xd403, 0x00, 0},
-	{ 0xd404, 0xbc, 0},
-	{ 0xd405, 0x23, 0},
-	{ 0xd406, 0x00, 0},
-	{ 0xd407, 0x01, 0},
-	{ 0xd408, 0x10, 0},
-	{ 0xd409, 0x00, 0},
-	{ 0xd40a, 0x00, 0},
-	{ 0xd40b, 0x25, 0},
-	{ 0xd40c, 0x9d, 0},
-	{ 0xd40d, 0x00, 0},
-	{ 0xd40e, 0x00, 0},
-	{ 0xd40f, 0x00, 0},
-	{ 0xd410, 0x00, 0},
-	{ 0xd411, 0x00, 0},
-	{ 0xd412, 0x00, 0},
-	{ 0xd413, 0x0b, 0},
-	{ 0xd414, 0xb8, 0},
-	{ 0xd415, 0xe8, 0},
-	{ 0xd416, 0x00, 0},
-	{ 0xd417, 0x02, 0},
-	{ 0xd418, 0x07, 0},
-	{ 0xd419, 0xff, 0},
-	{ 0xd41a, 0xd6, 0},
-	{ 0xd41b, 0x24, 0},
-	{ 0xd41c, 0x15, 0},
-	{ 0xd41d, 0x00, 0},
-	{ 0xd41e, 0x00, 0},
-	{ 0xd41f, 0x00, 0},
-	{ 0xd420, 0x18, 0},
-	{ 0xd421, 0x60, 0},
-	{ 0xd422, 0x80, 0},
-	{ 0xd423, 0x06, 0},
-	{ 0xd424, 0xa8, 0},
-	{ 0xd425, 0x63, 0},
-	{ 0xd426, 0xc4, 0},
-	{ 0xd427, 0xb8, 0},
-	{ 0xd428, 0x8c, 0},
-	{ 0xd429, 0x63, 0},
-	{ 0xd42a, 0x00, 0},
-	{ 0xd42b, 0x00, 0},
-	{ 0xd42c, 0xbc, 0},
-	{ 0xd42d, 0x23, 0},
-	{ 0xd42e, 0x00, 0},
-	{ 0xd42f, 0x01, 0},
-	{ 0xd430, 0x10, 0},
-	{ 0xd431, 0x00, 0},
-	{ 0xd432, 0x00, 0},
-	{ 0xd433, 0x1b, 0},
-	{ 0xd434, 0x9d, 0},
-	{ 0xd435, 0x00, 0},
-	{ 0xd436, 0x00, 0},
-	{ 0xd437, 0x00, 0},
-	{ 0xd438, 0xb8, 0},
-	{ 0xd439, 0xe8, 0},
-	{ 0xd43a, 0x00, 0},
-	{ 0xd43b, 0x02, 0},
-	{ 0xd43c, 0x9c, 0},
-	{ 0xd43d, 0xc0, 0},
-	{ 0xd43e, 0x00, 0},
-	{ 0xd43f, 0x00, 0},
-	{ 0xd440, 0x18, 0},
-	{ 0xd441, 0xa0, 0},
-	{ 0xd442, 0x80, 0},
-	{ 0xd443, 0x06, 0},
-	{ 0xd444, 0xe0, 0},
-	{ 0xd445, 0x67, 0},
-	{ 0xd446, 0x30, 0},
-	{ 0xd447, 0x00, 0},
-	{ 0xd448, 0xa8, 0},
-	{ 0xd449, 0xa5, 0},
-	{ 0xd44a, 0xce, 0},
-	{ 0xd44b, 0xb0, 0},
-	{ 0xd44c, 0x19, 0},
-	{ 0xd44d, 0x60, 0},
-	{ 0xd44e, 0x00, 0},
-	{ 0xd44f, 0x01, 0},
-	{ 0xd450, 0xa9, 0},
-	{ 0xd451, 0x6b, 0},
-	{ 0xd452, 0x06, 0},
-	{ 0xd453, 0x14, 0},
-	{ 0xd454, 0xe0, 0},
-	{ 0xd455, 0x83, 0},
-	{ 0xd456, 0x28, 0},
-	{ 0xd457, 0x00, 0},
-	{ 0xd458, 0x9c, 0},
-	{ 0xd459, 0xc6, 0},
-	{ 0xd45a, 0x00, 0},
-	{ 0xd45b, 0x01, 0},
-	{ 0xd45c, 0xe0, 0},
-	{ 0xd45d, 0x63, 0},
-	{ 0xd45e, 0x18, 0},
-	{ 0xd45f, 0x00, 0},
-	{ 0xd460, 0x8c, 0},
-	{ 0xd461, 0x84, 0},
-	{ 0xd462, 0x00, 0},
-	{ 0xd463, 0x00, 0},
-	{ 0xd464, 0xe0, 0},
-	{ 0xd465, 0xa3, 0},
-	{ 0xd466, 0x58, 0},
-	{ 0xd467, 0x00, 0},
-	{ 0xd468, 0xa4, 0},
-	{ 0xd469, 0xc6, 0},
-	{ 0xd46a, 0x00, 0},
-	{ 0xd46b, 0xff, 0},
-	{ 0xd46c, 0xb8, 0},
-	{ 0xd46d, 0x64, 0},
-	{ 0xd46e, 0x00, 0},
-	{ 0xd46f, 0x18, 0},
-	{ 0xd470, 0xbc, 0},
-	{ 0xd471, 0x46, 0},
-	{ 0xd472, 0x00, 0},
-	{ 0xd473, 0x03, 0},
-	{ 0xd474, 0x94, 0},
-	{ 0xd475, 0x85, 0},
-	{ 0xd476, 0x00, 0},
-	{ 0xd477, 0x00, 0},
-	{ 0xd478, 0xb8, 0},
-	{ 0xd479, 0x63, 0},
-	{ 0xd47a, 0x00, 0},
-	{ 0xd47b, 0x98, 0},
-	{ 0xd47c, 0xe0, 0},
-	{ 0xd47d, 0x64, 0},
-	{ 0xd47e, 0x18, 0},
-	{ 0xd47f, 0x00, 0},
-	{ 0xd480, 0x0f, 0},
-	{ 0xd481, 0xff, 0},
-	{ 0xd482, 0xff, 0},
-	{ 0xd483, 0xf0, 0},
-	{ 0xd484, 0xdc, 0},
-	{ 0xd485, 0x05, 0},
-	{ 0xd486, 0x18, 0},
-	{ 0xd487, 0x00, 0},
-	{ 0xd488, 0x9c, 0},
-	{ 0xd489, 0x68, 0},
-	{ 0xd48a, 0x00, 0},
-	{ 0xd48b, 0x01, 0},
-	{ 0xd48c, 0xa5, 0},
-	{ 0xd48d, 0x03, 0},
-	{ 0xd48e, 0x00, 0},
-	{ 0xd48f, 0xff, 0},
-	{ 0xd490, 0xbc, 0},
-	{ 0xd491, 0x48, 0},
-	{ 0xd492, 0x00, 0},
-	{ 0xd493, 0x01, 0},
-	{ 0xd494, 0x0f, 0},
-	{ 0xd495, 0xff, 0},
-	{ 0xd496, 0xff, 0},
-	{ 0xd497, 0xea, 0},
-	{ 0xd498, 0xb8, 0},
-	{ 0xd499, 0xe8, 0},
-	{ 0xd49a, 0x00, 0},
-	{ 0xd49b, 0x02, 0},
-	{ 0xd49c, 0x18, 0},
-	{ 0xd49d, 0x60, 0},
-	{ 0xd49e, 0x00, 0},
-	{ 0xd49f, 0x01, 0},
-	{ 0xd4a0, 0xa8, 0},
-	{ 0xd4a1, 0x63, 0},
-	{ 0xd4a2, 0x06, 0},
-	{ 0xd4a3, 0x14, 0},
-	{ 0xd4a4, 0x07, 0},
-	{ 0xd4a5, 0xff, 0},
-	{ 0xd4a6, 0xe4, 0},
-	{ 0xd4a7, 0x05, 0},
-	{ 0xd4a8, 0x9c, 0},
-	{ 0xd4a9, 0x83, 0},
-	{ 0xd4aa, 0x00, 0},
-	{ 0xd4ab, 0x10, 0},
-	{ 0xd4ac, 0x85, 0},
-	{ 0xd4ad, 0x21, 0},
-	{ 0xd4ae, 0x00, 0},
-	{ 0xd4af, 0x00, 0},
-	{ 0xd4b0, 0x44, 0},
-	{ 0xd4b1, 0x00, 0},
-	{ 0xd4b2, 0x48, 0},
-	{ 0xd4b3, 0x00, 0},
-	{ 0xd4b4, 0x9c, 0},
-	{ 0xd4b5, 0x21, 0},
-	{ 0xd4b6, 0x00, 0},
-	{ 0xd4b7, 0x04, 0},
-	{ 0xd4b8, 0x18, 0},
-	{ 0xd4b9, 0x60, 0},
-	{ 0xd4ba, 0x00, 0},
-	{ 0xd4bb, 0x01, 0},
-	{ 0xd4bc, 0x9c, 0},
-	{ 0xd4bd, 0x80, 0},
-	{ 0xd4be, 0xff, 0},
-	{ 0xd4bf, 0xff, 0},
-	{ 0xd4c0, 0xa8, 0},
-	{ 0xd4c1, 0x63, 0},
-	{ 0xd4c2, 0x09, 0},
-	{ 0xd4c3, 0xef, 0},
-	{ 0xd4c4, 0xd8, 0},
-	{ 0xd4c5, 0x03, 0},
-	{ 0xd4c6, 0x20, 0},
-	{ 0xd4c7, 0x00, 0},
-	{ 0xd4c8, 0x18, 0},
-	{ 0xd4c9, 0x60, 0},
-	{ 0xd4ca, 0x80, 0},
-	{ 0xd4cb, 0x06, 0},
-	{ 0xd4cc, 0xa8, 0},
-	{ 0xd4cd, 0x63, 0},
-	{ 0xd4ce, 0xc9, 0},
-	{ 0xd4cf, 0xef, 0},
-	{ 0xd4d0, 0xd8, 0},
-	{ 0xd4d1, 0x03, 0},
-	{ 0xd4d2, 0x20, 0},
-	{ 0xd4d3, 0x00, 0},
-	{ 0xd4d4, 0x44, 0},
-	{ 0xd4d5, 0x00, 0},
-	{ 0xd4d6, 0x48, 0},
-	{ 0xd4d7, 0x00, 0},
-	{ 0xd4d8, 0x15, 0},
-	{ 0xd4d9, 0x00, 0},
-	{ 0xd4da, 0x00, 0},
-	{ 0xd4db, 0x00, 0},
-	{ 0xd4dc, 0x18, 0},
-	{ 0xd4dd, 0x80, 0},
-	{ 0xd4de, 0x00, 0},
-	{ 0xd4df, 0x01, 0},
-	{ 0xd4e0, 0xa8, 0},
-	{ 0xd4e1, 0x84, 0},
-	{ 0xd4e2, 0x0a, 0},
-	{ 0xd4e3, 0x12, 0},
-	{ 0xd4e4, 0x8c, 0},
-	{ 0xd4e5, 0x64, 0},
-	{ 0xd4e6, 0x00, 0},
-	{ 0xd4e7, 0x00, 0},
-	{ 0xd4e8, 0xbc, 0},
-	{ 0xd4e9, 0x03, 0},
-	{ 0xd4ea, 0x00, 0},
-	{ 0xd4eb, 0x00, 0},
-	{ 0xd4ec, 0x13, 0},
-	{ 0xd4ed, 0xff, 0},
-	{ 0xd4ee, 0xff, 0},
-	{ 0xd4ef, 0xfe, 0},
-	{ 0xd4f0, 0x15, 0},
-	{ 0xd4f1, 0x00, 0},
-	{ 0xd4f2, 0x00, 0},
-	{ 0xd4f3, 0x00, 0},
-	{ 0xd4f4, 0x44, 0},
-	{ 0xd4f5, 0x00, 0},
-	{ 0xd4f6, 0x48, 0},
-	{ 0xd4f7, 0x00, 0},
-	{ 0xd4f8, 0x15, 0},
-	{ 0xd4f9, 0x00, 0},
-	{ 0xd4fa, 0x00, 0},
-	{ 0xd4fb, 0x00, 0},
-	{ 0xd4fc, 0x00, 0},
-	{ 0xd4fd, 0x00, 0},
-	{ 0xd4fe, 0x00, 0},
-	{ 0xd4ff, 0x00, 0},
-	{ 0xd500, 0x00, 0},
-	{ 0xd501, 0x00, 0},
-	{ 0xd502, 0x00, 0},
-	{ 0xd503, 0x00, 0},
+	{ 0xd356, 0x00, 0 },
+	{ 0xd357, 0x11, 0 },
+	{ 0xd358, 0xa8, 0 },
+	{ 0xd359, 0x83, 0 },
+	{ 0xd35a, 0x6e, 0 },
+	{ 0xd35b, 0x43, 0 },
+	{ 0xd35c, 0xe0, 0 },
+	{ 0xd35d, 0x6c, 0 },
+	{ 0xd35e, 0x28, 0 },
+	{ 0xd35f, 0x02, 0 },
+	{ 0xd360, 0xe0, 0 },
+	{ 0xd361, 0x84, 0 },
+	{ 0xd362, 0x28, 0 },
+	{ 0xd363, 0x02, 0 },
+	{ 0xd364, 0x07, 0 },
+	{ 0xd365, 0xff, 0 },
+	{ 0xd366, 0xf8, 0 },
+	{ 0xd367, 0x30, 0 },
+	{ 0xd368, 0xb8, 0 },
+	{ 0xd369, 0x63, 0 },
+	{ 0xd36a, 0x00, 0 },
+	{ 0xd36b, 0x08, 0 },
+	{ 0xd36c, 0x03, 0 },
+	{ 0xd36d, 0xff, 0 },
+	{ 0xd36e, 0xff, 0 },
+	{ 0xd36f, 0xc0, 0 },
+	{ 0xd370, 0x85, 0 },
+	{ 0xd371, 0x4e, 0 },
+	{ 0xd372, 0x00, 0 },
+	{ 0xd373, 0x00, 0 },
+	{ 0xd374, 0x03, 0 },
+	{ 0xd375, 0xff, 0 },
+	{ 0xd376, 0xff, 0 },
+	{ 0xd377, 0xe7, 0 },
+	{ 0xd378, 0xd4, 0 },
+	{ 0xd379, 0x01, 0 },
+	{ 0xd37a, 0x40, 0 },
+	{ 0xd37b, 0x18, 0 },
+	{ 0xd37c, 0x9c, 0 },
+	{ 0xd37d, 0x60, 0 },
+	{ 0xd37e, 0x00, 0 },
+	{ 0xd37f, 0x00, 0 },
+	{ 0xd380, 0x03, 0 },
+	{ 0xd381, 0xff, 0 },
+	{ 0xd382, 0xff, 0 },
+	{ 0xd383, 0xdb, 0 },
+	{ 0xd384, 0xd4, 0 },
+	{ 0xd385, 0x01, 0 },
+	{ 0xd386, 0x18, 0 },
+	{ 0xd387, 0x14, 0 },
+	{ 0xd388, 0x03, 0 },
+	{ 0xd389, 0xff, 0 },
+	{ 0xd38a, 0xff, 0 },
+	{ 0xd38b, 0xce, 0 },
+	{ 0xd38c, 0x9d, 0 },
+	{ 0xd38d, 0x6b, 0 },
+	{ 0xd38e, 0x00, 0 },
+	{ 0xd38f, 0xff, 0 },
+	{ 0xd390, 0x03, 0 },
+	{ 0xd391, 0xff, 0 },
+	{ 0xd392, 0xff, 0 },
+	{ 0xd393, 0xc6, 0 },
+	{ 0xd394, 0x9c, 0 },
+	{ 0xd395, 0x63, 0 },
+	{ 0xd396, 0x00, 0 },
+	{ 0xd397, 0xff, 0 },
+	{ 0xd398, 0xa8, 0 },
+	{ 0xd399, 0xe3, 0 },
+	{ 0xd39a, 0x38, 0 },
+	{ 0xd39b, 0xf , 0 },
+	{ 0xd39c, 0x8c, 0 },
+	{ 0xd39d, 0x84, 0 },
+	{ 0xd39e, 0x00, 0 },
+	{ 0xd39f, 0x00, 0 },
+	{ 0xd3a0, 0xa8, 0 },
+	{ 0xd3a1, 0xa3, 0 },
+	{ 0xd3a2, 0x38, 0 },
+	{ 0xd3a3, 0x0e, 0 },
+	{ 0xd3a4, 0xa8, 0 },
+	{ 0xd3a5, 0xc3, 0 },
+	{ 0xd3a6, 0x6e, 0 },
+	{ 0xd3a7, 0x42, 0 },
+	{ 0xd3a8, 0xd8, 0 },
+	{ 0xd3a9, 0x07, 0 },
+	{ 0xd3aa, 0x20, 0 },
+	{ 0xd3ab, 0x00, 0 },
+	{ 0xd3ac, 0x8c, 0 },
+	{ 0xd3ad, 0x66, 0 },
+	{ 0xd3ae, 0x00, 0 },
+	{ 0xd3af, 0x00, 0 },
+	{ 0xd3b0, 0xd8, 0 },
+	{ 0xd3b1, 0x05, 0 },
+	{ 0xd3b2, 0x18, 0 },
+	{ 0xd3b3, 0x00, 0 },
+	{ 0xd3b4, 0x85, 0 },
+	{ 0xd3b5, 0x21, 0 },
+	{ 0xd3b6, 0x00, 0 },
+	{ 0xd3b7, 0x00, 0 },
+	{ 0xd3b8, 0x85, 0 },
+	{ 0xd3b9, 0x41, 0 },
+	{ 0xd3ba, 0x00, 0 },
+	{ 0xd3bb, 0x04, 0 },
+	{ 0xd3bc, 0x85, 0 },
+	{ 0xd3bd, 0x81, 0 },
+	{ 0xd3be, 0x00, 0 },
+	{ 0xd3bf, 0x08, 0 },
+	{ 0xd3c0, 0x85, 0 },
+	{ 0xd3c1, 0xc1, 0 },
+	{ 0xd3c2, 0x00, 0 },
+	{ 0xd3c3, 0x0c, 0 },
+	{ 0xd3c4, 0x86, 0 },
+	{ 0xd3c5, 0x01, 0 },
+	{ 0xd3c6, 0x00, 0 },
+	{ 0xd3c7, 0x10, 0 },
+	{ 0xd3c8, 0x44, 0 },
+	{ 0xd3c9, 0x00, 0 },
+	{ 0xd3ca, 0x48, 0 },
+	{ 0xd3cb, 0x00, 0 },
+	{ 0xd3cc, 0x9c, 0 },
+	{ 0xd3cd, 0x21, 0 },
+	{ 0xd3ce, 0x00, 0 },
+	{ 0xd3cf, 0x1c, 0 },
+	{ 0xd3d0, 0x9c, 0 },
+	{ 0xd3d1, 0x21, 0 },
+	{ 0xd3d2, 0xff, 0 },
+	{ 0xd3d3, 0xfc, 0 },
+	{ 0xd3d4, 0xd4, 0 },
+	{ 0xd3d5, 0x01, 0 },
+	{ 0xd3d6, 0x48, 0 },
+	{ 0xd3d7, 0x00, 0 },
+	{ 0xd3d8, 0x18, 0 },
+	{ 0xd3d9, 0x60, 0 },
+	{ 0xd3da, 0x00, 0 },
+	{ 0xd3db, 0x01, 0 },
+	{ 0xd3dc, 0xa8, 0 },
+	{ 0xd3dd, 0x63, 0 },
+	{ 0xd3de, 0x07, 0 },
+	{ 0xd3df, 0x80, 0 },
+	{ 0xd3e0, 0x8c, 0 },
+	{ 0xd3e1, 0x63, 0 },
+	{ 0xd3e2, 0x00, 0 },
+	{ 0xd3e3, 0x68, 0 },
+	{ 0xd3e4, 0xbc, 0 },
+	{ 0xd3e5, 0x03, 0 },
+	{ 0xd3e6, 0x00, 0 },
+	{ 0xd3e7, 0x00, 0 },
+	{ 0xd3e8, 0x10, 0 },
+	{ 0xd3e9, 0x00, 0 },
+	{ 0xd3ea, 0x00, 0 },
+	{ 0xd3eb, 0x0c, 0 },
+	{ 0xd3ec, 0x15, 0 },
+	{ 0xd3ed, 0x00, 0 },
+	{ 0xd3ee, 0x00, 0 },
+	{ 0xd3ef, 0x00, 0 },
+	{ 0xd3f0, 0x07, 0 },
+	{ 0xd3f1, 0xff, 0 },
+	{ 0xd3f2, 0xd9, 0 },
+	{ 0xd3f3, 0x98, 0 },
+	{ 0xd3f4, 0x15, 0 },
+	{ 0xd3f5, 0x00, 0 },
+	{ 0xd3f6, 0x00, 0 },
+	{ 0xd3f7, 0x00, 0 },
+	{ 0xd3f8, 0x18, 0 },
+	{ 0xd3f9, 0x60, 0 },
+	{ 0xd3fa, 0x80, 0 },
+	{ 0xd3fb, 0x06, 0 },
+	{ 0xd3fc, 0xa8, 0 },
+	{ 0xd3fd, 0x63, 0 },
+	{ 0xd3fe, 0xc4, 0 },
+	{ 0xd3ff, 0xb8, 0 },
+	{ 0xd400, 0x8c, 0 },
+	{ 0xd401, 0x63, 0 },
+	{ 0xd402, 0x00, 0 },
+	{ 0xd403, 0x00, 0 },
+	{ 0xd404, 0xbc, 0 },
+	{ 0xd405, 0x23, 0 },
+	{ 0xd406, 0x00, 0 },
+	{ 0xd407, 0x01, 0 },
+	{ 0xd408, 0x10, 0 },
+	{ 0xd409, 0x00, 0 },
+	{ 0xd40a, 0x00, 0 },
+	{ 0xd40b, 0x25, 0 },
+	{ 0xd40c, 0x9d, 0 },
+	{ 0xd40d, 0x00, 0 },
+	{ 0xd40e, 0x00, 0 },
+	{ 0xd40f, 0x00, 0 },
+	{ 0xd410, 0x00, 0 },
+	{ 0xd411, 0x00, 0 },
+	{ 0xd412, 0x00, 0 },
+	{ 0xd413, 0x0b, 0 },
+	{ 0xd414, 0xb8, 0 },
+	{ 0xd415, 0xe8, 0 },
+	{ 0xd416, 0x00, 0 },
+	{ 0xd417, 0x02, 0 },
+	{ 0xd418, 0x07, 0 },
+	{ 0xd419, 0xff, 0 },
+	{ 0xd41a, 0xd6, 0 },
+	{ 0xd41b, 0x24, 0 },
+	{ 0xd41c, 0x15, 0 },
+	{ 0xd41d, 0x00, 0 },
+	{ 0xd41e, 0x00, 0 },
+	{ 0xd41f, 0x00, 0 },
+	{ 0xd420, 0x18, 0 },
+	{ 0xd421, 0x60, 0 },
+	{ 0xd422, 0x80, 0 },
+	{ 0xd423, 0x06, 0 },
+	{ 0xd424, 0xa8, 0 },
+	{ 0xd425, 0x63, 0 },
+	{ 0xd426, 0xc4, 0 },
+	{ 0xd427, 0xb8, 0 },
+	{ 0xd428, 0x8c, 0 },
+	{ 0xd429, 0x63, 0 },
+	{ 0xd42a, 0x00, 0 },
+	{ 0xd42b, 0x00, 0 },
+	{ 0xd42c, 0xbc, 0 },
+	{ 0xd42d, 0x23, 0 },
+	{ 0xd42e, 0x00, 0 },
+	{ 0xd42f, 0x01, 0 },
+	{ 0xd430, 0x10, 0 },
+	{ 0xd431, 0x00, 0 },
+	{ 0xd432, 0x00, 0 },
+	{ 0xd433, 0x1b, 0 },
+	{ 0xd434, 0x9d, 0 },
+	{ 0xd435, 0x00, 0 },
+	{ 0xd436, 0x00, 0 },
+	{ 0xd437, 0x00, 0 },
+	{ 0xd438, 0xb8, 0 },
+	{ 0xd439, 0xe8, 0 },
+	{ 0xd43a, 0x00, 0 },
+	{ 0xd43b, 0x02, 0 },
+	{ 0xd43c, 0x9c, 0 },
+	{ 0xd43d, 0xc0, 0 },
+	{ 0xd43e, 0x00, 0 },
+	{ 0xd43f, 0x00, 0 },
+	{ 0xd440, 0x18, 0 },
+	{ 0xd441, 0xa0, 0 },
+	{ 0xd442, 0x80, 0 },
+	{ 0xd443, 0x06, 0 },
+	{ 0xd444, 0xe0, 0 },
+	{ 0xd445, 0x67, 0 },
+	{ 0xd446, 0x30, 0 },
+	{ 0xd447, 0x00, 0 },
+	{ 0xd448, 0xa8, 0 },
+	{ 0xd449, 0xa5, 0 },
+	{ 0xd44a, 0xce, 0 },
+	{ 0xd44b, 0xb0, 0 },
+	{ 0xd44c, 0x19, 0 },
+	{ 0xd44d, 0x60, 0 },
+	{ 0xd44e, 0x00, 0 },
+	{ 0xd44f, 0x01, 0 },
+	{ 0xd450, 0xa9, 0 },
+	{ 0xd451, 0x6b, 0 },
+	{ 0xd452, 0x06, 0 },
+	{ 0xd453, 0x14, 0 },
+	{ 0xd454, 0xe0, 0 },
+	{ 0xd455, 0x83, 0 },
+	{ 0xd456, 0x28, 0 },
+	{ 0xd457, 0x00, 0 },
+	{ 0xd458, 0x9c, 0 },
+	{ 0xd459, 0xc6, 0 },
+	{ 0xd45a, 0x00, 0 },
+	{ 0xd45b, 0x01, 0 },
+	{ 0xd45c, 0xe0, 0 },
+	{ 0xd45d, 0x63, 0 },
+	{ 0xd45e, 0x18, 0 },
+	{ 0xd45f, 0x00, 0 },
+	{ 0xd460, 0x8c, 0 },
+	{ 0xd461, 0x84, 0 },
+	{ 0xd462, 0x00, 0 },
+	{ 0xd463, 0x00, 0 },
+	{ 0xd464, 0xe0, 0 },
+	{ 0xd465, 0xa3, 0 },
+	{ 0xd466, 0x58, 0 },
+	{ 0xd467, 0x00, 0 },
+	{ 0xd468, 0xa4, 0 },
+	{ 0xd469, 0xc6, 0 },
+	{ 0xd46a, 0x00, 0 },
+	{ 0xd46b, 0xff, 0 },
+	{ 0xd46c, 0xb8, 0 },
+	{ 0xd46d, 0x64, 0 },
+	{ 0xd46e, 0x00, 0 },
+	{ 0xd46f, 0x18, 0 },
+	{ 0xd470, 0xbc, 0 },
+	{ 0xd471, 0x46, 0 },
+	{ 0xd472, 0x00, 0 },
+	{ 0xd473, 0x03, 0 },
+	{ 0xd474, 0x94, 0 },
+	{ 0xd475, 0x85, 0 },
+	{ 0xd476, 0x00, 0 },
+	{ 0xd477, 0x00, 0 },
+	{ 0xd478, 0xb8, 0 },
+	{ 0xd479, 0x63, 0 },
+	{ 0xd47a, 0x00, 0 },
+	{ 0xd47b, 0x98, 0 },
+	{ 0xd47c, 0xe0, 0 },
+	{ 0xd47d, 0x64, 0 },
+	{ 0xd47e, 0x18, 0 },
+	{ 0xd47f, 0x00, 0 },
+	{ 0xd480, 0xf , 0 },
+	{ 0xd481, 0xff, 0 },
+	{ 0xd482, 0xff, 0 },
+	{ 0xd483, 0xf0, 0 },
+	{ 0xd484, 0xdc, 0 },
+	{ 0xd485, 0x05, 0 },
+	{ 0xd486, 0x18, 0 },
+	{ 0xd487, 0x00, 0 },
+	{ 0xd488, 0x9c, 0 },
+	{ 0xd489, 0x68, 0 },
+	{ 0xd48a, 0x00, 0 },
+	{ 0xd48b, 0x01, 0 },
+	{ 0xd48c, 0xa5, 0 },
+	{ 0xd48d, 0x03, 0 },
+	{ 0xd48e, 0x00, 0 },
+	{ 0xd48f, 0xff, 0 },
+	{ 0xd490, 0xbc, 0 },
+	{ 0xd491, 0x48, 0 },
+	{ 0xd492, 0x00, 0 },
+	{ 0xd493, 0x01, 0 },
+	{ 0xd494, 0xf , 0 },
+	{ 0xd495, 0xff, 0 },
+	{ 0xd496, 0xff, 0 },
+	{ 0xd497, 0xea, 0 },
+	{ 0xd498, 0xb8, 0 },
+	{ 0xd499, 0xe8, 0 },
+	{ 0xd49a, 0x00, 0 },
+	{ 0xd49b, 0x02, 0 },
+	{ 0xd49c, 0x18, 0 },
+	{ 0xd49d, 0x60, 0 },
+	{ 0xd49e, 0x00, 0 },
+	{ 0xd49f, 0x01, 0 },
+	{ 0xd4a0, 0xa8, 0 },
+	{ 0xd4a1, 0x63, 0 },
+	{ 0xd4a2, 0x06, 0 },
+	{ 0xd4a3, 0x14, 0 },
+	{ 0xd4a4, 0x07, 0 },
+	{ 0xd4a5, 0xff, 0 },
+	{ 0xd4a6, 0xe4, 0 },
+	{ 0xd4a7, 0x05, 0 },
+	{ 0xd4a8, 0x9c, 0 },
+	{ 0xd4a9, 0x83, 0 },
+	{ 0xd4aa, 0x00, 0 },
+	{ 0xd4ab, 0x10, 0 },
+	{ 0xd4ac, 0x85, 0 },
+	{ 0xd4ad, 0x21, 0 },
+	{ 0xd4ae, 0x00, 0 },
+	{ 0xd4af, 0x00, 0 },
+	{ 0xd4b0, 0x44, 0 },
+	{ 0xd4b1, 0x00, 0 },
+	{ 0xd4b2, 0x48, 0 },
+	{ 0xd4b3, 0x00, 0 },
+	{ 0xd4b4, 0x9c, 0 },
+	{ 0xd4b5, 0x21, 0 },
+	{ 0xd4b6, 0x00, 0 },
+	{ 0xd4b7, 0x04, 0 },
+	{ 0xd4b8, 0x18, 0 },
+	{ 0xd4b9, 0x60, 0 },
+	{ 0xd4ba, 0x00, 0 },
+	{ 0xd4bb, 0x01, 0 },
+	{ 0xd4bc, 0x9c, 0 },
+	{ 0xd4bd, 0x80, 0 },
+	{ 0xd4be, 0xff, 0 },
+	{ 0xd4bf, 0xff, 0 },
+	{ 0xd4c0, 0xa8, 0 },
+	{ 0xd4c1, 0x63, 0 },
+	{ 0xd4c2, 0x09, 0 },
+	{ 0xd4c3, 0xef, 0 },
+	{ 0xd4c4, 0xd8, 0 },
+	{ 0xd4c5, 0x03, 0 },
+	{ 0xd4c6, 0x20, 0 },
+	{ 0xd4c7, 0x00, 0 },
+	{ 0xd4c8, 0x18, 0 },
+	{ 0xd4c9, 0x60, 0 },
+	{ 0xd4ca, 0x80, 0 },
+	{ 0xd4cb, 0x06, 0 },
+	{ 0xd4cc, 0xa8, 0 },
+	{ 0xd4cd, 0x63, 0 },
+	{ 0xd4ce, 0xc9, 0 },
+	{ 0xd4cf, 0xef, 0 },
+	{ 0xd4d0, 0xd8, 0 },
+	{ 0xd4d1, 0x03, 0 },
+	{ 0xd4d2, 0x20, 0 },
+	{ 0xd4d3, 0x00, 0 },
+	{ 0xd4d4, 0x44, 0 },
+	{ 0xd4d5, 0x00, 0 },
+	{ 0xd4d6, 0x48, 0 },
+	{ 0xd4d7, 0x00, 0 },
+	{ 0xd4d8, 0x15, 0 },
+	{ 0xd4d9, 0x00, 0 },
+	{ 0xd4da, 0x00, 0 },
+	{ 0xd4db, 0x00, 0 },
+	{ 0xd4dc, 0x18, 0 },
+	{ 0xd4dd, 0x80, 0 },
+	{ 0xd4de, 0x00, 0 },
+	{ 0xd4df, 0x01, 0 },
+	{ 0xd4e0, 0xa8, 0 },
+	{ 0xd4e1, 0x84, 0 },
+	{ 0xd4e2, 0x0a, 0 },
+	{ 0xd4e3, 0x12, 0 },
+	{ 0xd4e4, 0x8c, 0 },
+	{ 0xd4e5, 0x64, 0 },
+	{ 0xd4e6, 0x00, 0 },
+	{ 0xd4e7, 0x00, 0 },
+	{ 0xd4e8, 0xbc, 0 },
+	{ 0xd4e9, 0x03, 0 },
+	{ 0xd4ea, 0x00, 0 },
+	{ 0xd4eb, 0x00, 0 },
+	{ 0xd4ec, 0x13, 0 },
+	{ 0xd4ed, 0xff, 0 },
+	{ 0xd4ee, 0xff, 0 },
+	{ 0xd4ef, 0xfe, 0 },
+	{ 0xd4f0, 0x15, 0 },
+	{ 0xd4f1, 0x00, 0 },
+	{ 0xd4f2, 0x00, 0 },
+	{ 0xd4f3, 0x00, 0 },
+	{ 0xd4f4, 0x44, 0 },
+	{ 0xd4f5, 0x00, 0 },
+	{ 0xd4f6, 0x48, 0 },
+	{ 0xd4f7, 0x00, 0 },
+	{ 0xd4f8, 0x15, 0 },
+	{ 0xd4f9, 0x00, 0 },
+	{ 0xd4fa, 0x00, 0 },
+	{ 0xd4fb, 0x00, 0 },
+	{ 0xd4fc, 0x00, 0 },
+	{ 0xd4fd, 0x00, 0 },
+	{ 0xd4fe, 0x00, 0 },
+	{ 0xd4ff, 0x00, 0 },
+	{ 0xd500, 0x00, 0 },
+	{ 0xd501, 0x00, 0 },
+	{ 0xd502, 0x00, 0 },
+	{ 0xd503, 0x00, 0 },
 	{ 0x6f0e, 0x33, 0 },
 	{ 0x6f0f, 0x33, 0 },
 	{ 0x460e, 0x08, 0 },
@@ -1730,7 +1730,7 @@ enum ov10635_frame_rate {
 	{ 0x4611, 0x01, 0 },
 	{ 0x4612, 0x00, 0 },
 	{ 0x4613, 0x01, 0 },
-	{ 0x4605, 0x08, 0},
+	{ 0x4605, 0x08, 0 },
 	{ 0x4608, 0x00, 0 },
 	{ 0x4609, 0x08, 0 },
 	{ 0x6804, 0x00, 0 },
@@ -1740,11 +1740,11 @@ enum ov10635_frame_rate {
 	{ 0x3510, 0x00, 0 },
 	{ 0x3504, 0x00, 0 },
 	{ 0x6800, 0x00, 0 },
-	{ 0x6f0d, 0x0f, 0 },
+	{ 0x6f0d, 0xf , 0 },
 	{ 0x5000, 0xff, 0 },
 	{ 0x5001, 0xbf, 0 },
 	{ 0x5002, 0x7e, 0 },
-	{ 0x5003, 0x0c, 0},
+	{ 0x5003, 0x0c, 0 },
 	{ 0x503d, 0x00, 0 },
 	{ 0xc450, 0x01, 0 },
 	{ 0xc452, 0x04, 0 },
@@ -1818,7 +1818,7 @@ enum ov10635_frame_rate {
 	{ 0xc4a9, 0x80, 0 },
 	{ 0xc4aa, 0x0d, 0 },
 	{ 0xc4ab, 0x00, 0 },
-	{ 0xc4ac, 0x0f, 0 },
+	{ 0xc4ac, 0xf , 0 },
 	{ 0xc4ad, 0xc0, 0 },
 	{ 0xc4b4, 0x01, 0 },
 	{ 0xc4b5, 0x01, 0 },
@@ -1868,7 +1868,7 @@ enum ov10635_frame_rate {
 	{ 0xc4f8, 0x04, 0 },
 	{ 0xc4f9, 0x0b, 0 },
 	{ 0xc4fa, 0x00, 0 },
-	{ 0xc4fb, 0x00, 0},
+	{ 0xc4fb, 0x00, 0 },
 	{ 0xc4fc, 0x01, 0 },
 	{ 0xc4fd, 0x00, 0 },
 	{ 0xc4fe, 0x04, 0 },
@@ -2035,7 +2035,6 @@ enum ov10635_frame_rate {
 	{ 0x301b, 0xf0, 0 },
 	{ 0x301c, 0xf0, 0 },
 	{ 0x301a, 0xf0, 0 },
-
 	{ 0xceb0, 0x00, 0 },
 	{ 0xceb1, 0x00, 0 },
 	{ 0xceb2, 0x00, 0 },
@@ -2047,68 +2046,6 @@ enum ov10635_frame_rate {
 	{ 0xc4bc, 0x01, 0 },
 	{ 0xc4bd, 0x60, 0 },
 
-	{ 0x301b, 0xf1, 0 }, //PLL
-	{ 0x3003, 0x18, 0 },
-	{ 0x3004, 0x11, 0 },
-	{ 0x3005, 0x12, 0 },
-	{ 0x3006, 0x11, 0 },
-	{ 0x3024, 0x00, 0 },
-	{ 0x3621, 0x63, 0 },
-	{ 0x3702, 0x20, 0 },
-	{ 0x3703, 0x48, 0 },
-	{ 0x3704, 0x32, 0 },
-	{ 0x3800, 0x00, 0 },
-	{ 0x3801, 0x00, 0 },
-	{ 0x3802, 0x00, 0 },
-	{ 0x3803, 0x04, 0 },
-	{ 0x3804, 0x00, 0 },
-	{ 0x3805, 0xFF, 0 },
-	{ 0x3806, 0x03, 0 },
-	{ 0x3807, 0x29, 0 },
-	{ 0x3808, 0x05, 0 },
-	{ 0x3809, 0x00, 0 },
-	{ 0x380a, 0x03, 0 },
-	{ 0x380b, 0x20, 0 },
-	{ 0x380c, 0x07, 0 },
-	{ 0x380d, 0x70, 0 },
-	{ 0x6e42, 0x03, 0 },
-	{ 0x6e43, 0x48, 0 },
-	{ 0x3810, 0x00, 0 },
-	{ 0x3811, 0x10, 0 },
-	{ 0x3812, 0x00, 0 },
-	{ 0x3813, 0x02, 0 },
-	{ 0x381c, 0x00, 0 },
-	{ 0x381e, 0x00, 0 },
-	{ 0x381f, 0x0C, 0 },
-	{ 0x4001, 0x06, 0 },
-	{ 0x4004, 0x04, 0 },
-	{ 0x4050, 0x22, 0 },
-	{ 0x4051, 0x24, 0 },
-	{ 0x4605, 0x0C, 0 },
-	{ 0x4606, 0x0B, 0 },
-	{ 0x4607, 0x28, 0 },
-	{ 0xc488, 0x34, 0 },
-	{ 0xc489, 0x00, 0 },
-	{ 0xc48a, 0x34, 0 },
-	{ 0xc48b, 0x00, 0 },
-	{ 0xc4cc, 0x10, 0 },
-	{ 0xc4cd, 0x18, 0 },
-	{ 0xc4ce, 0x10, 0 },
-	{ 0xc4cf, 0x18, 0 },
-	{ 0xc510, 0x00, 0 },
-	{ 0xc511, 0x01, 0 },
-	{ 0xc512, 0x01, 0 },
-	{ 0xc513, 0x88, 0 },
-	{ 0x5005, 0x08, 0 },
-	{ 0x3007, 0x01, 0 },
-	{ 0xc518, 0x03, 0 },
-	{ 0xc519, 0x48, 0 },
-	{ 0xc51a, 0x07, 0 },
-	{ 0xc51b, 0x70, 0 },
-	{ 0x5608, 0x0D, 0 },
-	{ 0x3815, 0x8C, 0 },
-	{ 0x301b, 0xf0, 0 },
-
 	{ 0x4709, 0x10, 0 },/* dvp swap */
 	{ 0x4300, 0x3a, 0 },/* YUV order UYVY */
 	{ 0x3832, 0x01, 0 },/* fsin */
-- 
1.7.9.5

